--- usr/www/avme/en/html/en/fon/fonab.frm	2008-01-01 19:00:00.000000000 -0500
+++ usr/www/avme/en/html/en/fon/fonab.frm	2008-02-14 03:48:12.000000000 -0500
@@ -0,0 +1,10 @@
+<input type="hidden" name="tam:settings/TAM0/Active" value="<? query tam:settings/TAM0/Active ?>" id="uiPostAktiv">
+<input type="hidden" name="tam:settings/TAM0/MSNBitmap" value="<? query tam:settings/TAM0/MSNBitmap ?>" id="uiPostBitmap">
+<? multiquery sip:settings/sip/list(displayname) '
+<input type="hidden" name="sip:settings/$1/displayname" value="$2" id="ui$1" disabled>
+' ?>
+<? repeat [ 0 9 ] `
+<input type="hidden" name="telcfg:settings/MSN/MSN$0" value="<? query telcfg:settings/MSN/MSN$0 ?>" id="uiMsn$0" disabled>` ?>
+<input type="hidden" name="telcfg:settings/MSN/POTS" value="<? query telcfg:settings/MSN/POTS ?>" id="uiPots" disabled>
+<? repeat [ 0 9 ] `
+<input type="hidden" name="tam:settings/MSN$0" value="<? query tam:settings/MSN$0 ?>" id="uiPostTamMsn$0">` ?>
--- usr/www/avme/en/html/en/fon/fonab.html	2008-01-01 19:00:00.000000000 -0500
+++ usr/www/avme/en/html/en/fon/fonab.html	2008-02-14 03:48:12.000000000 -0500
@@ -0,0 +1,35 @@
+<? include ../html/$var:lang/fon/fonab.inc ?>
+<div id="content">
+<div class="backtitel"><div class="rundrt"><div class="rundlt"><div class="ecklb"><div class="eckrb"><div class="foretitel">
+<? echo $var:Text1 ?>
+</div></div></div></div></div></div>
+<div class="backdialog"><div class="ecklm"><div class="eckrm"><div class="ecklb"><div class="eckrb"><div class="foredialog">
+<div style="padding-left: 20px">
+<p><input type="checkbox" id="uiViewAktiv">&nbsp;<label for="uiViewAktiv"><? echo $var:Text2 ?></label></p>
+<p style="padding-top:10px"><label for="uiViewN00"><? echo $var:Text3 ?></label></p>
+<p class="combo">
+<select size="1" id="uiViewN00" class="Eingabefeld" style="width:200px"></select>
+</p>
+<p style="padding-top:10px"><label for="uiViewN01"><? echo $var:Text4 ?></label></p>
+<table>
+<tr>
+<td style="vertical-align:top">
+<p class="combo"><select size="1" id="uiViewN01" class="Eingabefeld" style="width:200px"></select></p>
+<p class="combo"><select size="1" id="uiViewN02" class="Eingabefeld" style="width:200px" onchange="OnChange(id, value)"></select></p>
+<p class="combo" id="p03" style="display:none"><select size="1" id="uiViewN03" class="Eingabefeld" style="width:200px" onchange="OnChange(id, value)"></select></p>
+<p class="combo" id="p04" style="display:none"><select size="1" id="uiViewN04" class="Eingabefeld" style="width:200px" onchange="OnChange(id, value)"></select></p>
+<p class="combo" id="p05" style="display:none"><select size="1" id="uiViewN05" class="Eingabefeld" style="width:200px" onchange="OnChange(id, value)"></select></p>
+</td>
+<td style="vertical-align:top">
+<p class="combo" id="p06" style="display:none"><select size="1" id="uiViewN06" class="Eingabefeld" style="width:200px" onchange="OnChange(id, value)"></select></p>
+<p class="combo" id="p07" style="display:none"><select size="1" id="uiViewN07" class="Eingabefeld" style="width:200px" onchange="OnChange(id, value)"></select></p>
+<p class="combo" id="p08" style="display:none"><select size="1" id="uiViewN08" class="Eingabefeld" style="width:200px" onchange="OnChange(id, value)"></select></p>
+<p class="combo" id="p09" style="display:none"><select size="1" id="uiViewN09" class="Eingabefeld" style="width:200px" onchange="OnChange(id, value)"></select></p>
+</td>
+</tr>
+</table>
+</div>
+<? include ../html/$var:lang/home/error.html ?>
+</div></div></div></div></div></div>
+<? include ../html/$var:lang/home/rbb_applycancel.html ?>
+</div>
--- usr/www/avme/en/html/en/fon/fonab.inc	2008-01-01 19:00:00.000000000 -0500
+++ usr/www/avme/en/html/en/fon/fonab.inc	2008-02-14 03:48:12.000000000 -0500
@@ -0,0 +1,5 @@
+<? include ../html/$var:lang/global.inc ?>
+<? setvariable var:Text1 'Answering machine ' ?>
+<? setvariable var:Text2 'Activate answering machine' ?>
+<? setvariable var:Text3 'Phonenumber in use for answering machine' ?>
+<? setvariable var:Text4 'additional Phonenumbers' ?>
--- usr/www/avme/en/html/en/fon/fonab.js	2008-01-01 19:00:00.000000000 -0500
+++ usr/www/avme/en/html/en/fon/fonab.js	2008-02-14 03:48:12.000000000 -0500
@@ -0,0 +1,119 @@
+<!--		../fon/fonab.js		-->
+<style type="text/css">
+<!--
+#content {width: 540px; margin: auto;}
+.pFett {font-weight: bold; 	padding-bottom: 10px;}
+.combo {padding-top: 2px; padding-left: 20px;}
+.left20 {margin-left: 20px;}
+-->
+</style>
+<script type="text/javascript">
+<? include ../html/$var:lang/js/jsl.js ?>
+var g_txtFestnetz	= "Festnetz: ";
+var g_txtInternet	= "Internet: ";
+<? multiquery sip:settings/sip/list '
+var g_$1;
+' ?>
+var g_msn0;
+var g_msn1;
+var g_msn2;
+var g_msn3;
+var g_msn4;
+var g_msn5;
+var g_msn6;
+var g_msn7;
+var g_msn8;
+var g_msn9;
+var g_Pots;
+function uiDoOnLoad() {
+<? multiquery sip:settings/sip/list '
+g_$1 = jslGetValue("ui$1");
+' ?>
+g_msn0 = jslGetValue("uiMsn0");
+g_msn1 = jslGetValue("uiMsn1");
+g_msn2 = jslGetValue("uiMsn2");
+g_msn3 = jslGetValue("uiMsn3");
+g_msn4 = jslGetValue("uiMsn4");
+g_msn5 = jslGetValue("uiMsn5");
+g_msn6 = jslGetValue("uiMsn6");
+g_msn7 = jslGetValue("uiMsn7");
+g_msn8 = jslGetValue("uiMsn8");
+g_msn9 = jslGetValue("uiMsn9");
+g_Pots = jslGetValue("uiPots");
+fillId("uiViewN00", jslGetValue("uiPostTamMsn0"));
+fillId("uiViewN01", jslGetValue("uiPostTamMsn1"));
+fillId("uiViewN02", jslGetValue("uiPostTamMsn2"));
+var i, j;
+for (i = 9; i >1 && jslGetValue("uiPostTamMsn"+i) == ""; i--) ;
+for (j = Math.min(9, i+1); j>2; j--) {
+jslDisplay("p0"+j, true);
+fillId("uiViewN0"+j, jslGetValue("uiPostTamMsn"+j));
+}
+jslSetChecked("uiViewAktiv", jslGetValue("uiPostAktiv") == "1");
+}
+function uiDoSave() {
+jslSetValue("uiPostAktiv", jslGetChecked("uiViewAktiv") ? "1" :"0");
+<? repeat [ 0 9 ] `
+if (jslGetValue("uiViewN0$0") == "Leer") jslSetValue("uiPostTamMsn$0", ""); else jslCopyValue("uiPostTamMsn$0", "uiViewN0$0");` ?>
+jslSetValue("uiPostBitmap", "1023");
+jslSubmitFormEx("fon", "fonlist");
+}
+function uiDoCancel() {
+jslGoTo("fon", "fonlist");
+}
+function uiDoHelp() {
+jslPopHelp("hilfe_fon_nebenstelle");
+}
+function fillId(id, nr) {
+<? if eq '$var:usePSTN' 1 `
+// Auswahl mit den vorgegebenen MSN füllen und Eintrag msn selektieren
+<? repeat [ 0 9 ] `
+if (g_msn$10 != "") addOption(id, g_msn$10, g_txtFestnetz+g_msn$10);` ?>
+if (g_Pots != "") addOption(id, g_Pots, g_txtFestnetz+g_Pots);
+` ?> 
+// Sip Zugänge anfügen
+<? multiquery sip:settings/sip/list(ID,displayname) '
+if ("$3" != "") addOption(id, "SIP"+"$2", g_txtInternet+g_$1);
+' ?>
+addOption(id, "Leer", "");
+jslSetSelection(id, nr=="" ? "Leer":nr);
+}
+function addMsn (id, msn) {
+if (msn != "") addOption(id, msn, g_txtFestnetz+msn);
+}
+function addOption (id, val, text) {
+var element;
+element = document.getElementById(id);
+if (element == null) return;
+var eintrag = new Option(text, val, false, false);
+element.options[element.length]=eintrag;
+}
+function isFilled(id) {
+var element = document.getElementById(id);
+if (element == null) return false;
+return (element.length>0);
+}
+function enableSelect(num) {
+jslDisplay("p"+num, true);
+if (!isFilled("uiViewN"+num))
+fillId("uiViewN"+num, "");
+}
+function OnChange (id, value) {
+if (value == "Leer") return;
+switch (id) {
+case "uiViewN02": enableSelect("03"); break;
+case "uiViewN03": enableSelect("04"); break;
+case "uiViewN04": enableSelect("05"); break;
+case "uiViewN05": enableSelect("06"); break;
+case "uiViewN06": enableSelect("07"); break;
+case "uiViewN07": enableSelect("08"); break;
+case "uiViewN08": enableSelect("09"); break;
+}
+}
+function SetChecked (postid, viewid) {
+jslSetChecked(viewid, jslGetValue(postid) != "0");
+}
+function GetChecked (postid, viewid) {
+jslSetValue(postid, jslGetChecked(viewid) ? "1":"0");
+}
+</script>
--- usr/www/avme/en/html/en/fon/tam.frm	2008-01-01 19:00:00.000000000 -0500
+++ usr/www/avme/en/html/en/fon/tam.frm	2008-02-14 03:48:12.000000000 -0500
@@ -0,0 +1,5 @@
+<input type="hidden" name="tam:settings/TAM0/Active" value="<? query tam:settings/TAM0/Active ?>" id="uiPostActive" />
+<input type="hidden" name="tam:settings/TAM0/Mode" value="<? query tam:settings/TAM0/Mode ?>" id="uiPostMode" />
+<input type="hidden" name="tam:settings/TAM0/RingCount" value="<? query tam:settings/TAM0/RingCount ?>" id="uiPostRing" />
+<input type="hidden" name="tam:settings/TAM0/RecordLength" value="<? query tam:settings/TAM0/RecordLength ?>" id="uiPostLen" />
+<input type="hidden" name="tam:settings/TAM0/PIN" value="<? query tam:settings/TAM0/PIN ?>" id="uiPostPin" />
--- usr/www/avme/en/html/en/fon/tam.html	2008-01-01 19:00:00.000000000 -0500
+++ usr/www/avme/en/html/en/fon/tam.html	2008-02-14 03:48:12.000000000 -0500
@@ -0,0 +1,53 @@
+<? include ../html/$var:lang/fon/tam.inc ?>
+<div id="content">
+<div class="backtitel"><div class="rundrt"><div class="rundlt"><div class="ecklb"><div class="eckrb"><div class="foretitel">
+<? echo $var:Text1 ?>
+</div></div></div></div></div></div>
+<div class="backdialog"><div class="ecklm"><div class="eckrm"><div class="ecklb"><div class="eckrb"><div class="foredialog">
+<p class="mt10 mb10 fett" >Answering Machine configuration</p>
+<p class="mb5"><input type="checkbox" id="uiViewActive" onclick="OnActiv()">&nbsp;<label for="uiViewActive"><? echo $var:Text2 ?></label></p>
+<table class="ml25">
+<tr>
+<td><label for="uiViewMode"><? echo $var:Text3 ?></label></td>
+<td><select id="uiViewMode" size="1" class="Eingabefeld" onchange="OnChangeMode(value)">
+<option value="0"><? echo $var:Text4 ?></option>
+<option value="1"><? echo $var:Text5 ?></option>
+</select></td>
+</tr>
+<tr>
+<td><label for="uiViewRing"><? echo $var:Text6 ?></label></td>
+<td><select size="1" id="uiViewRing" class="Eingabefeld">
+<option value="0"><? echo $var:Text7 ?></option>
+<option value="1">1</option>
+<option value="2">2</option>
+<option value="3">3</option>
+<option value="4">4</option>
+<option value="5">5</option>
+<option value="6">6</option>
+<option value="255">automatic</option>
+</select></td>
+</tr>
+<tr>
+<td><label for="uiViewLen"><? echo $var:Text8 ?></label></td>
+<td><select size="1" id="uiViewLen" class="Eingabefeld">
+<option value="0">No timelimit</option>
+<option value="60">60 Secunds</option>
+<option value="120">120 Secunds</option>
+<option value="180">180 Secunds</option>
+</select></td>
+</tr>
+</table>
+<p class="mt10 mb5"><? echo $var:Text10 ?></p>
+<table class="ml25">
+<tr>
+<td><label for="uiViewPin"><? echo $var:Text11 ?></label></td>
+<td><input type="password" size="12" maxlength="4" id="uiViewPin" onfocus="uiSelect(id);" /></td>
+</tr>
+<tr>
+<td><label for="uiViewPin2"><? echo $var:Text12 ?></label></td>
+<td><input type="password" size="12" maxlength="4" id="uiViewPin2" onfocus="uiSelect(id);" /></td>
+</tr>
+</table>
+</div></div></div></div></div></div>
+<? include ../html/$var:lang/home/rbb_applycancel.html ?>
+</div>
--- usr/www/avme/en/html/en/fon/tam.js	2008-01-01 19:00:00.000000000 -0500
+++ usr/www/avme/en/html/en/fon/tam.js	2008-02-14 03:48:12.000000000 -0500
@@ -0,0 +1,62 @@
+<!--		../fon/tam.js		-->
+<style type="text/css">
+<!--
+#content {width: 480px; margin: auto;}
+td {padding: 3px 5px;}
+.ml25 {margin-left: 25px;}
+-->
+</style>
+<script type="text/javascript">
+<? include ../html/$var:lang/js/val.js ?>
+<? include ../html/$var:lang/js/jsl.js ?>
+var g_mldPin 	= "for PIN use numeral input only.";
+var g_mldPinLength = "PIN must be 4 digits.";
+var g_mldPinDiff = "PIN doesn´t match your confirmation.";
+function uiDoOnLoad() {
+jslSetChecked("uiViewActive", jslGetValue("uiPostActive") == "1");
+jslSetSelection("uiViewMode", jslGetValue("uiPostMode"));
+jslSetSelection("uiViewRing", jslGetValue("uiPostRing"));
+jslSetSelection("uiViewLen", jslGetValue("uiPostLen"));
+jslCopyValue("uiViewPin", "uiPostPin");
+jslCopyValue("uiViewPin2", "uiPostPin");
+OnActiv();
+}
+function uiDoSave() {
+if (jslGetChecked("uiViewActive")) {
+jslSetValue("uiPostActive", "1");
+jslCopyValue("uiPostMode","uiViewMode");
+jslCopyValue("uiPostRing","uiViewRing");
+
+var p = jslGetValue("uiViewPin");
+if (!valIsZahlVorhanden(p)) {alert(g_mldPin); return;}
+if (p.length != 4) {alert(g_mldPinLength); return;}
+
+var p2 = jslGetValue("uiViewPin2");
+if (p2!=p) {alert(g_mldPinDiff); return;}
+
+jslCopyValue("uiPostLen", "uiViewLen");
+jslCopyValue("uiPostPin", "uiViewPin");
+} else {
+jslSetValue("uiPostActive", "0");
+}
+
+jslFormSubmitEx("fon", "tam");
+}
+function uiDoCancel() {
+jslGoTo("fon", "tam");
+}
+function uiDoHelp() {
+jslPopHelp("hilfe_fon_anrufbeantworter");
+}
+function OnActiv() {
+var b = jslGetChecked("uiViewActive");
+jslSetEnabled("uiViewMode", b);
+jslSetEnabled("uiViewRing", b);
+jslSetEnabled("uiViewLen", b && (jslGetValue("uiViewMode")=="1"));
+jslSetEnabled("uiViewPin", b);
+jslSetEnabled("uiViewPin2", b);
+}
+function OnChangeMode (value) {
+jslSetEnabled("uiViewLen", value=="1");
+}
+</script>
--- usr/www/avme/en/html/en/fon/tam.inc	2008-04-18 18:09:54.000000000 -0400
+++ usr/www/avme/en/html/en/fon/tam.inc	2008-04-18 18:09:31.000000000 -0400
@@ -0,0 +1,13 @@
+<? include ../html/$var:lang/global.inc ?>
+<? setvariable var:Text1 'Answering machine' ?>
+<? setvariable var:Text2 'activate answering machine' ?>
+<? setvariable var:Text3 'Mode' ?>
+<? setvariable var:Text4 'message only' ?>
+<? setvariable var:Text5 'record message' ?>
+<? setvariable var:Text6 'alert character' ?>
+<? setvariable var:Text7 'immediate call' ?>
+<? setvariable var:Text8 'recording lenght' ?>
+<? setvariable var:Text9 'seconds' ?>
+<? setvariable var:Text10 'If you like to access your answering machine from a differnt phone line you need a 4 digits PIN.' ?>
+<? setvariable var:Text11 'PIN' ?>
+<? setvariable var:Text12 'Confirm PIN' ?>
