--- usr/www/avme/en/html/en/fon/dect0.frm	2008-01-01 19:00:00.000000000 -0500
+++ usr/www/avme/en/html/en/fon/dect0.frm	2008-02-17 12:15:49.000000000 -0500
@@ -0,0 +1,32 @@
+<input type="hidden" name="var:FonNr" value="<? echo $var:FonNr ?>" id="uiPostFonNr">
+<input type="hidden" name="var:IsdnNr" value="<? echo $var:IsdnNr ?>" id="uiPostIsdnNr">
+<input type="hidden" name="var:parent" value="<? echo $var:parent ?>" id="uiPostParent">
+<input type="hidden" name="var:ShowIsdnDefault" value="<? echo $var:ShowIsdnDefault ?>" id="uiShowIsdnDefault">
+<input type="hidden" name="telcfg:settings/MSN/POTS" value="<? query telcfg:settings/MSN/POTS ?>" id="uiPostPots">
+<input type="hidden" name="telcfg:settings/MSN/NTDefault" value="<? query telcfg:settings/MSN/NTDefault ?>" id="uiPostNTDefault">
+<input type="hidden" name="telcfg:settings/MSN/MSN0" value="<? query telcfg:settings/MSN/MSN0 ?>" id="uiPostMsn0" disabled>
+<input type="hidden" name="telcfg:settings/MSN/MSN1" value="<? query telcfg:settings/MSN/MSN1 ?>" id="uiPostMsn1" disabled>
+<input type="hidden" name="telcfg:settings/MSN/MSN2" value="<? query telcfg:settings/MSN/MSN2 ?>" id="uiPostMsn2" disabled>
+<input type="hidden" name="telcfg:settings/MSN/MSN3" value="<? query telcfg:settings/MSN/MSN3 ?>" id="uiPostMsn3" disabled>
+<input type="hidden" name="telcfg:settings/MSN/MSN4" value="<? query telcfg:settings/MSN/MSN4 ?>" id="uiPostMsn4" disabled>
+<input type="hidden" name="telcfg:settings/MSN/MSN5" value="<? query telcfg:settings/MSN/MSN5 ?>" id="uiPostMsn5" disabled>
+<input type="hidden" name="telcfg:settings/MSN/MSN6" value="<? query telcfg:settings/MSN/MSN6 ?>" id="uiPostMsn6" disabled>
+<input type="hidden" name="telcfg:settings/MSN/MSN7" value="<? query telcfg:settings/MSN/MSN7 ?>" id="uiPostMsn7" disabled>
+<input type="hidden" name="telcfg:settings/MSN/MSN8" value="<? query telcfg:settings/MSN/MSN8 ?>" id="uiPostMsn8" disabled>
+<input type="hidden" name="telcfg:settings/MSN/MSN9" value="<? query telcfg:settings/MSN/MSN9 ?>" id="uiPostMsn9" disabled>
+<? multiquery sip:settings/sip/list(displayname) '
+<input type="hidden" name="sip:settings/$1/displayname" value="$2" id="uiPost$1" disabled>
+' ?>
+<input type="hidden" name="telcfg:settings/MSN/Port3/BusyOnBusy" value="<? query telcfg:settings/MSN/Port3/BusyOnBusy ?>" id="uiPostBusy">
+<input type="hidden" name="telcfg:settings/MSN/Port3/CallWaitingProt" value="<? query telcfg:settings/MSN/Port3/CallWaitingProt ?>" id="uiPostWaiting">
+<input type="hidden" name="telcfg:settings/MSN/Port3/NoRingWithNightSetting" value="<? query telcfg:settings/MSN/Port3/NoRingWithNightSetting?>" id="uiPostNoRingNS">
+<input type="hidden" name="telcfg:settings/MSN/Port3/RingAllowed" value="<? query telcfg:settings/MSN/Port3/RingAllowed ?>" id="uiPostRingAllowed0">
+<input type="hidden" name="telcfg:settings/MSN/Port3/NoRingTime" value="<? query telcfg:settings/MSN/Port3/NoRingTime?>" id="uiPostNoRingTime0">
+<input type="hidden" name="telcfg:settings/MSN/Port3/COLR" value="<? query telcfg:settings/MSN/Port3/COLR?>" id="uiPostColr">
+<input type="hidden" name="telcfg:settings/MSN/Port3/MWI_Voice" value="<? query telcfg:settings/MSN/Port3/MWI_Voice?>" id="uiPostMwiVoice">
+<input type="hidden" name="telcfg:settings/MSN/Port3/MWI_Fax" value="<? query telcfg:settings/MSN/Port3/MWI_Fax?>" id="uiPostMwiFax">
+<input type="hidden" name="telcfg:settings/MSN/Port3/MWI_Mail" value="<? query telcfg:settings/MSN/Port3/MWI_Mail?>" id="uiPostMwiMail">
+<input type="hidden" name="telcfg:settings/MSN/Port3/MWI_Once" value="<? query telcfg:settings/MSN/Port3/MWI_Once ?>" id="uiPostMwiOnce">
+
+<input type="hidden" name="dect:command/PageHandsets" id="uiPostPageHandsets" disabled>
+<input type="hidden" name="dect:command/DateTime" id="uiPostHandsetsTime" disabled>
--- usr/www/avme/en/html/en/fon/dect0.html	2008-01-01 19:00:00.000000000 -0500
+++ usr/www/avme/en/html/en/fon/dect0.html	2008-03-04 05:55:29.000000000 -0500
@@ -0,0 +1,47 @@
+<? include ../html/$var:lang/fon/dect0.inc ?>
+<div id="content">
+<div class="backtitel"><div class="rundrt"><div class="rundlt"><div class="ecklb"><div class="eckrb"><div class="foretitel">
+<? echo '$var:Text23' ?>
+</div></div></div></div></div></div>
+
+<div class="backframe"><div class="ecklm"><div class="eckrm"><div class="rundrb"><div class="rundlb"><div class="foredialog"><div class="ie_b10">
+<ul class="tabs">
+<li class="tabs_on"><a href="javascript:uiDoRefresh()"><? echo '$var:Text24' ?></a></li>
+<li><a href="javascript:uiDoListDect()"><? echo '$var:Text26' ?></a></li>
+<li><a href="javascript:uiDoSetupDect()"><? echo '$var:Text28' ?></a></li>
+<li><a href="javascript:uiDoEditDect()"><? echo '$var:Text29' ?></a></li>
+</ul>
+<div class="backdialog"><div class="ecklm"><div class="eckrm"><div class="ecklb"><div class="eckrb"><div class="foredialog">
+<p class="mb10"><? echo $var:Text3 ?></p>
+<p class="ml25 mb5"><? echo $var:Text4 ?></p>
+<p class="ml50 mb10"><select size="1" id="uiViewNr" class="Eingabefeld" style="width:300px"></select></p>
+<p class="mb10"><? echo $var:Text5 ?></p>
+<p class="ml25 mb5"><? echo $var:Text6 ?></p>
+<p class="mb5">&nbsp;</p>
+<table id="t1" class="ml50 tborder" style="width: 300px">
+<? multiquery sip:settings/sip/list(displayname) `
+<tr id="uiSip$0" style="display:none"><td class="c1">$var:Text16</td><td class="c2">$2</td></tr>
+` ?>
+<? repeat [ 0 9 ] `
+<tr id="uiMsn$0" style="display:none"><td class="c1">$var:Text7</td><td class="c2"><? query telcfg:settings/MSN/MSN$0 ?></td></tr>
+` ?>
+<tr id="uiPots" style="display:none"><td class="c1"><? echo $var:Text7 ?></td><td class="c2"><? query telcfg:settings/MSN/POTS ?></td></tr>
+<tr id="uiNichts" style="display:none"><td colspan=2 class="ac"><? echo $var:Text17 ?></td></tr>
+</table>
+</div></div></div></div></div></div>
+
+<div class="backdialog"><div class="ecklm"><div class="eckrm"><div class="ecklb"><div class="eckrb"><div class="foredialog">
+<p class="mb5"><? echo $var:Text35 ?></p>
+<div style="text-align: right;">
+<input type="button" value="<? echo $var:Text33 ?>" id="uiViewPage" onclick="uiDoPage()" class="Pushbutton" style="width:180px;">
+<input type="button" value="<? echo $var:Text34 ?>" id="uiViewDateTime" onclick="uiDoDateTime()" class="Pushbutton" style="width:180px;">
+<input type="button" value="<? echo $var:Text19 ?>" id="uiSendBook" onclick="uiDoSendBook()" class="Pushbutton" style="width:180px;">
+</div>
+</div></div></div></div></div></div>
+
+<? include ../html/$var:lang/home/rbb_applycancel.html ?>
+</div></div></div></div></div></div></div>
+</div>
+
+
+
--- usr/www/avme/en/html/en/fon/dect0.inc	2008-01-01 19:00:00.000000000 -0500
+++ usr/www/avme/en/html/en/fon/dect0.inc	2008-03-05 10:05:17.000000000 -0500
@@ -0,0 +1,21 @@
+<? include ../html/$var:lang/global.inc ?>
+<? setvariable var:Text1 'DECT-Devices' ?>
+<? setvariable var:Text3 'DECT-Devices without MSN assinged place calls via the mail phonenumber and take all incoming calls.' ?>
+<? setvariable var:Text4 'Main Phonenumber' ?>
+<? setvariable var:Text5 'If you like you can configure the following MSN to your DECT devices. Details can be found in the manual of your DECT device.' ?>
+<? setvariable var:Text6 'Phonenumbers' ?>
+<? setvariable var:Text7 'Fixed subsriber line:' ?>
+<? setvariable var:Text16 'Internet:' ?>
+<? setvariable var:Text17 'No phonenumber configured' ?>
+<? setvariable var:Text19 'Send Phonebook' ?>
+<? setvariable var:Text23 'DECT-devices' ?>
+<? setvariable var:Text24 'Overview' ?>
+<? setvariable var:Text25 'Fixed subsriber line' ?>
+<? setvariable var:Text26 'Internal phonenumbers' ?>
+<? setvariable var:Text28 'Phonenumber mappings' ?>
+<? setvariable var:Text29 'Mobile devices' ?>
+<? setvariable var:Text30 'Additional devices' ?>
+<? setvariable var:Text32 'No mobile devices assigned' ?>
+<? setvariable var:Text33 'Page mobile device' ?>
+<? setvariable var:Text34 'Set date and time' ?>
+<? setvariable var:Text35 'With the Paging-Function you can locate your mobile devices. At the push of a Button you can also set Time and Date of the DECT devices.' ?>
--- usr/www/avme/en/html/en/fon/dect0.js	2008-01-01 19:00:00.000000000 -0500
+++ usr/www/avme/en/html/en/fon/dect0.js	2008-03-04 06:21:25.000000000 -0500
@@ -0,0 +1,165 @@
+<!--		../fon/dect0.js		-->
+<style type="text/css">
+<!--
+#content {width: 580px; margin: auto;}
+.c1 {width: 150px; margin-left: 5px; padding: 2px 5px;}
+.c2 { margin-left: 5px; padding: 2px 0px;}
+.ml25 {margin-left: 25px;}
+.ml50 {margin-left: 50px;}
+.ar {text-align: right;}
+.ac {text-align: center;}
+#tMerkmale .c3 {padding: 4px 5px; text-align: center; vertical-align: top;}
+#tMerkmale .c4 {padding: 5px 5px;}
+.ml25 {margin-left: 25px;}
+.mt10 {margin-top: 10px;}
+-->
+</style>
+<script type="text/javascript">
+<? include ../html/$var:lang/js/val.js ?>
+<? include ../html/$var:lang/js/jsl.js ?>
+
+
+var g_txtInternettyp	= "Internet: ";
+var g_txtFestnetztyp	= "Festnetz: ";
+var g_mldDigit			= "Eine Rufnummer darf nur aus Ziffern bestehen.";
+
+<? multiquery sip:settings/sip/list '
+var g_$1;
+' ?>
+<? if eq '$var:usePSTN' 1 `
+var g_msn0;
+var g_msn1;
+var g_msn2;
+var g_msn3;
+var g_msn4;
+var g_msn5;
+var g_msn6;
+var g_msn7;
+var g_msn8;
+var g_msn9;
+` ?>
+function uiDoOnLoad() {
+<? multiquery sip:settings/sip/list '
+g_$1 = jslGetValue("uiPost$1");
+' ?>
+<? if eq '$var:usePSTN' 1 `
+g_msn0 = jslGetValue("uiPostMsn0");
+g_msn1 = jslGetValue("uiPostMsn1");
+g_msn2 = jslGetValue("uiPostMsn2");
+g_msn3 = jslGetValue("uiPostMsn3");
+g_msn4 = jslGetValue("uiPostMsn4");
+g_msn5 = jslGetValue("uiPostMsn5");
+g_msn6 = jslGetValue("uiPostMsn6");
+g_msn7 = jslGetValue("uiPostMsn7");
+g_msn8 = jslGetValue("uiPostMsn8");
+g_msn9 = jslGetValue("uiPostMsn9");
+` ?>
+fillId("uiViewNr", jslGetValue("uiPostNTDefault"));
+var bFound = false;
+<? multiquery sip:settings/sip/list(displayname) `
+if (valIsZahlVorhanden("$2")) {jslDisplay("uiSip$0", true); bFound = true;}
+` ?>
+<? if eq '$var:usePSTN' 1 `
+for (var i=0; i<10; i++) {
+if (jslGetValue("uiPostMsn"+i) != "") {jslDisplay("uiMsn"+i, true); bFound = true;}
+}
+if (jslGetValue("uiPostPots") != "") {jslDisplay("uiPots", true); bFound = true;}
+` ?>
+if (!bFound) jslDisplay("uiNichts", true);
+jslZebrastreifen("t1","#FFFFFF","#F8F8F8");
+<? setvariable var:handsetfound 0 ?>
+<? multiquery dect:settings/Handset/list(Subscribed,Type) `
+<? if eq $2 1 `
+<? if eq $3 1 `
+<? setvariable var:handsetfound 1 ?>
+` ?>
+` ?>
+` ?>
+jslDisplay("uiSendBook" , ("<? echo $var:handsetfound ?>" == "1"));
+jslSetValue("uiPostPageMaster", "dect0");
+jslSetValue("uiPostParent", "dect0");
+}
+function uiDoSave() {
+var nr = jslGetValue("uiViewNr");
+jslSetValue("uiPostNTDefault", nr == "Leer" ? "":nr);
+jslFormSubmitEx("fon", "dect0");
+}
+
+var g_menu = "<? print $var:menu ?>";
+
+function uiDoHelp() {
+jslPopHelp("hilfe_fon_dect");
+}
+function uiDoCancel() {
+jslGoTo(g_menu, "");
+}
+function uiDoRefresh() {
+jslGoTo(g_menu, "dect0");
+}
+function uiDoEditDect() {
+jslGoTo(g_menu, "foneditdect");
+}
+function uiDoSetupDect() {
+jslGoTo(g_menu, "fonsetupdect");
+}
+function uiDoListDect() {
+jslGoTo(g_menu, "fonlistdect");
+}
+function fillId(id, nr) {
+<? multiquery sip:settings/sip/list(ID,displayname) '
+if (valIsZahlVorhanden("$3")) addOption(id, "SIP"+"$2", g_txtInternettyp+g_$1);
+' ?>
+<? if eq '$var:usePSTN' 1 `
+addMsn(id, g_msn0);
+addMsn(id, g_msn1);
+addMsn(id, g_msn2);
+addMsn(id, g_msn3);
+addMsn(id, g_msn4);
+addMsn(id, g_msn5);
+addMsn(id, g_msn6);
+addMsn(id, g_msn7);
+addMsn(id, g_msn8);
+addMsn(id, g_msn9);
+addOption(id, "POTS", g_txtFestnetztyp+jslGetValue("uiPostPots"));
+` ?>
+addOption(id, "Leer", "");
+jslSetSelection(id, "Leer"); // Defaultwert
+jslSetSelection(id, nr=="" ? "Leer":nr);
+}
+function addMsn (id, msn) {
+if (msn != "") addOption(id, msn, g_txtFestnetztyp+msn);
+}
+function addOption (id, val, text) {
+var element;
+element = document.getElementById(id);
+if (element == null) return;
+var eintrag = new Option(text, val, false, false);
+element.options[element.length]=eintrag;
+}
+function SetChecked (postid, viewid) {
+jslSetChecked(viewid, jslGetValue(postid) != "0");
+}
+function GetChecked (postid, viewid) {
+jslSetValue(postid, jslGetChecked(viewid) ? "1":"0");
+}
+function OnOnce(n) {
+jslSetChecked("uiViewEinmal", n==0);
+jslSetChecked("uiViewImmer", n==1);
+}
+function uiDoSendBook() {
+jslSetValue("uiPostPageMaster", "dect0");
+jslGoTo("fon", "buchsend");
+}
+function uiDoPage () {
+jslSetValue("uiPostPageHandsets", "0");
+jslEnable("uiPostPageHandsets");
+jslSubmitFormEx("fon", "", "");
+if (!confirm("Alle angemeldeten Handteile sollten jetzt eine Minute lang klingeln!")) return;
+}
+function uiDoDateTime () {
+jslSetValue("uiPostHandsetsTime", "0");
+jslEnable("uiPostHandsetsTime");
+jslSubmitFormEx("fon", "", "");
+if (!confirm("Uhrzeit wurde nur bei allen angemldeten Swiss-Voice comaptiblen Handteilen gesetzt!")) return;
+}
+</script>
--- usr/www/avme/en/html/en/fon/fon1dect.frm	2008-01-01 19:00:00.000000000 -0500
+++ usr/www/avme/en/html/en/fon/fon1dect.frm	2008-02-14 13:34:20.000000000 -0500
@@ -0,0 +1,18 @@
+<input type="hidden" name="dect:settings/MSN<? echo $var:MsnNr ?>/MSN" value="<? query dect:settings/MSN<? echo $var:MsnNr ?>/MSN ?>" id="uiPostThisMSN">
+<input type="hidden" name="dect:settings/MSN<? echo $var:MsnNr ?>/CLIR" value="<? query dect:settings/MSN<? echo $var:MsnNr ?>/CLIR ?>" id="uiPostCLIR">
+<input type="hidden" name="dect:settings/MSN<? echo $var:MsnNr ?>/CallWaiting" value="<? query dect:settings/MSN<? echo $var:MsnNr ?>/CallWaiting ?>" id="uiPostCallWaiting">
+<input type="hidden" name="dect:settings/MSN<? echo $var:MsnNr ?>/BusyOnBusy" value="<? query dect:settings/MSN<? echo $var:MsnNr ?>/BusyOnBusy ?>" id="uiPostBusyOnBusy">
+<input type="hidden" name="dect:settings/MSN<? echo $var:MsnNr ?>/Handsets" value="<? query dect:settings/MSN<? echo $var:MsnNr ?>/Handsets ?>" id="uiPostHandsets">
+<? repeat [ 1 8 ] `
+<input type="hidden" name="telcfg:settings/NTHotDialList/Number$0" value="<? query telcfg:settings/NTHotDialList/Number$0 ?>" id="uiInt$0" disabled>
+` ?>
+<? multiquery sip:settings/sip/list(displayname) '
+<input type="hidden" name="sip:settings/$1/displayname" value="$2" id="ui$1" disabled>
+' ?>
+<? repeat [ 0 9 ] `
+<input type="hidden" name="telcfg:settings/MSN/MSN$0" value="<? query telcfg:settings/MSN/MSN$0 ?>" id="uiMsn$0" disabled>
+` ?>
+<input type="hidden" name="telcfg:settings/MSN/POTS" value="<? query telcfg:settings/MSN/POTS ?>" id="uiPots" disabled>
+<? repeat [ 0 4 ] `
+<input type="hidden" name="dect:settings/MSN$0/MSN" value="<? query dect:settings/MSN$0/MSN ?>" id="uiPostMSN$0" disabled>
+` ?>
--- usr/www/avme/en/html/en/fon/fon1dect.html	2008-01-01 19:00:00.000000000 -0500
+++ usr/www/avme/en/html/en/fon/fon1dect.html	2008-02-14 03:52:15.000000000 -0500
@@ -0,0 +1,41 @@
+<? include ../html/$var:lang/fon/fon1dect.inc ?>
+<div id="content">
+<div class="backtitel"><div class="rundrt"><div class="rundlt"><div class="ecklb"><div class="eckrb"><div class="foretitel">
+<? echo $var:Text1 ?>
+</div></div></div></div></div></div>
+<div class="backdialog"><div class="ecklm"><div class="eckrm"><div class="ecklb"><div class="eckrb"><div class="foredialog">
+<!--	<p><? echo $var:Text2 ?><script type="text/javascript">document.write(uiIndexMSN("<? echo $var:MsnNr ?>"));</script></p> -->	 
+<table>
+<tr>
+<td class="c1"><label for="uiViewMSN"><? echo $var:Text2 ?><script type="text/javascript">document.write(uiIndexMSN("<? echo $var:MsnNr ?>"));</script></label></td>
+<td><select size="1" id="uiViewMSN" class="Eingabefeld" style="width:200px"></select></td>
+</tr>
+<tr>
+<td class="c1"><label for="uiViewUser"><? echo $var:Text5 ?></label></td>
+<td><input type="text" size="33" maxlength="32" class="Eingabefeld" id="uiViewUser"></td>
+</tr>
+</table>
+<p><? echo $var:Text3 ?></p>
+<table>
+<tr>
+<td style="{text-align: left; width: 150px;"><input type="checkbox" name="uiViewCLIR" id="uiViewCLIR0">&nbsp;<label for="uiViewCLIR"><? echo $var:Text11 ?></label></td>
+<td style="{text-align: left; width: 150px;"><input type="checkbox" name="uiViewCallWaiting" id="uiViewCallWaiting">&nbsp;<label for="uiViewCallWaiting"><? echo $var:Text12 ?></label></td>
+<td style="{text-align: left; width: 150px;"><input type="checkbox" name="uiViewBusyOnBusy" id="uiViewBusyOnBusy">&nbsp;<label for="uiViewBusyOnBusy"><? echo $var:Text13 ?></label></td>
+</tr>
+</table>
+<p><? echo $var:Text4 ?></p>
+<table>
+<tr>
+<td style="{text-align: left; width: 150px;"><input type="checkbox" name="uiViewHandset0" id="uiViewHandset0" disabled>&nbsp;<label for="uiViewHandset0"><? echo $var:Text15 ?></label></td>
+<td style="{text-align: left; width: 150px;"><input type="checkbox" name="uiViewHandset2" id="uiViewHandset2" disabled>&nbsp;<label for="uiViewHandset2"><? echo $var:Text17 ?></label></td>
+<td style="{text-align: left; width: 150px;"><input type="checkbox" name="uiViewHandset4" id="uiViewHandset4" disabled>&nbsp;<label for="uiViewHandset4"><? echo $var:Text19 ?></label></td>
+</tr>
+<tr>
+<td style="{text-align: left; width: 150px;"><input type="checkbox" name="uiViewHandset1" id="uiViewHandset1" disabled>&nbsp;<label for="uiViewHandset1"><? echo $var:Text16 ?></label></td>
+<td style="{text-align: left; width: 150px;"><input type="checkbox" name="uiViewHandset3" id="uiViewHandset3" disabled>&nbsp;<label for="uiViewHandset3"><? echo $var:Text18 ?></label></td>
+<td style="{text-align: left; width: 150px;"><input type="checkbox" name="uiViewHandset5" id="uiViewHandset5" disabled>&nbsp;<label for="uiViewHandset5"><? echo $var:Text20 ?></label></td>
+</tr>
+</table>
+</div></div></div></div></div></div>
+<? include ../html/$var:lang/home/rbb_applycancel.html ?>
+</div>
--- usr/www/avme/en/html/en/fon/fon1dect.inc	2008-01-01 19:00:00.000000000 -0500
+++ usr/www/avme/en/html/en/fon/fon1dect.inc	2008-02-14 12:40:07.000000000 -0500
@@ -0,0 +1,17 @@
+<? include ../html/$var:lang/global.inc ?>
+<? setvariable var:Text1 'Assign MSN' ?>
+<? setvariable var:Text2 'MSN ' ?>
+<? setvariable var:Text3 'Options' ?>
+<? setvariable var:Text4 'Assign' ?>
+<? setvariable var:Text5 'Username' ?>
+<? setvariable var:Text11 'Call anonymous' ?>
+<? setvariable var:Text12 'knocking' ?>
+<? setvariable var:Text13 'MSN busy' ?>
+<? setvariable var:Text15 'Mobile device 1' ?>
+<? setvariable var:Text16 'Mobile device 2' ?>
+<? setvariable var:Text17 'Mobile device 3' ?>
+<? setvariable var:Text18 'Mobile device 4' ?>
+<? setvariable var:Text19 'Mobile device 5' ?>
+<? setvariable var:Text20 'Mobile device 6' ?>
+<? setvariable var:Text28 'Assign' ?>
+<? setvariable var:Text29 'Mobile devices' ?>
--- usr/www/avme/en/html/en/fon/fon1dect.js	2008-01-01 19:00:00.000000000 -0500
+++ usr/www/avme/en/html/en/fon/fon1dect.js	2008-03-05 09:57:21.000000000 -0500
@@ -0,0 +1,177 @@
+<!--		../fon/fon1dect.js		-->
+<style type="text/css">
+<!--
+#content {width: 540px; margin: auto;}
+td {padding: 5px 5px;}
+th {padding: 5px 5px;}
+-->
+</style>
+<script type="text/javascript">
+<? include ../html/$var:lang/js/val.js ?>
+<? include ../html/$var:lang/js/jsl.js ?>
+var g_txt_Festnetz	= "Fixed subscriber line: ";
+var g_txt_Intern	= "Internal: ";
+var g_txt_Internet	= "Internet: ";
+<? multiquery sip:settings/sip/list '
+var g_$1;
+' ?>
+var g_msn0;
+var g_msn1;
+var g_msn2;
+var g_msn3;
+var g_msn4;
+var g_msn5;
+var g_msn6;
+var g_msn7;
+var g_msn8;
+var g_msn9;
+var g_pots;
+var g_int1;
+var g_int2;
+var g_int3;
+var g_int4;
+var g_int5;
+var g_int6;
+var g_int7;
+var g_int8;
+var g_MsnGesperrt = new Array(0, 0, 0, 0, 0);
+var g_Index = new Array("A", "B", "C", "D", "E");
+var g_Handsets;
+var g_MsnNr;
+var g_OptsCount = 0;
+var g_mldNoNummer = "Please enter a Phonenumber.";
+var g_mldNoHandset = "Please select a mobile device.";
+var g_mldNoOpts = "All phonenumbers are already assigned.";
+function uiDoOnLoad() {
+g_Handsets = 0;
+
+<? multiquery dect:settings/Handset/list(Subscribed) `
+<? if eq $2 1 `
+g_Handsets |= (1<<$0);
+` ?>
+` ?>
+
+<? multiquery sip:settings/sip/list '
+g_$1 = jslGetValue("ui$1");
+' ?>
+g_msn0 = jslGetValue("uiMsn0");
+g_msn1 = jslGetValue("uiMsn1");
+g_msn2 = jslGetValue("uiMsn2");
+g_msn3 = jslGetValue("uiMsn3");
+g_msn4 = jslGetValue("uiMsn4");
+g_msn5 = jslGetValue("uiMsn5");
+g_msn6 = jslGetValue("uiMsn6");
+g_msn7 = jslGetValue("uiMsn7");
+g_msn8 = jslGetValue("uiMsn8");
+g_msn9 = jslGetValue("uiMsn9");
+g_pots = jslGetValue("uiPots");
+g_int1 = jslGetValue("uiInt1");
+g_int2 = jslGetValue("uiInt2");
+g_int3 = jslGetValue("uiInt3");
+g_int4 = jslGetValue("uiInt4");
+g_int5 = jslGetValue("uiInt5");
+g_int6 = jslGetValue("uiInt6");
+g_int7 = jslGetValue("uiInt7");
+g_int8 = jslGetValue("uiInt8");
+
+fillId("uiViewMSN", jslGetValue("uiPostThisMSN"));
+if (g_OptsCount == 1) {alert(g_mldNoOpts); return;}
+
+jslCopyValue("uiViewUser", jslGetValue("uiPostThisUser"));
+
+jslSetChecked("uiViewCLIR", jslGetValue("uiPostCLIR") == "1");
+jslSetChecked("uiViewCallWaiting", jslGetValue("uiPostCallWaiting") == "1");
+jslSetChecked("uiViewBusyOnBusy", jslGetValue("uiPostBusyOnBusy") == "1");
+
+var bits = jslGetValue("uiPostHandsets");
+var bits = (bits.charCodeAt(0) - 48);
+for (var j=0; j<6; j++) {
+if ((g_Handsets & (1<<j)) != 0)	{
+jslEnable("uiViewHandset"+j);
+} 
+jslSetChecked("uiViewHandset"+j, (bits & g_Handsets & (1<<j)) != 0);
+}
+}
+function uiDoSave() {
+if (jslGetValue("uiViewMSN") == "Leer") {alert(g_mldNoNummer); return;}
+
+var bits=0
+for (var j=0; j<6; j++) {
+if (jslGetChecked("uiViewHandset"+j)) bits |= 1<<j;
+}
+if (bits == 0) {alert(g_mldNoHandset); return;}
+
+jslCopyValue("uiPostThisMSN", "uiViewMSN");
+jslCopyValue("uiPostThisUser", "uiViewUser");
+jslSetValue("uiPostHandsets", bits);
+
+jslSetValue("uiPostCLIR", jslGetChecked("uiViewCLIR") ? "1":"0");
+jslSetValue("uiPostCallWaiting", jslGetChecked("uiViewCallWaiting") ? "1":"0");
+jslSetValue("uiPostBusyOnBusy", jslGetChecked("uiViewBusyOnBusy") ? "1":"0");
+
+jslSubmitFormEx("fon", jslGetValue("uiPostPageMaster"));
+}
+function uiDoCancel() {
+jslGoTo("fon", jslGetValue("uiPostPageMaster"));
+}
+function uiDoHelp() {
+jslPopHelp("hilfe_fon_fon1dect");
+}
+function isFrei(msn) {
+for (var i=0; i<5; i++) {
+if (i != g_MsnNr && msn == jslGetValue("uiPostMSN"+i)) return false;
+}
+return true;
+} 
+function fillId(id, nr) {
+<? if eq '$var:usePSTN' 1 `
+// Auswahl mit den vorgegebenen MSN füllen und Eintrag msn selektieren
+addMsn(id, g_msn0, g_txt_Festnetz);
+addMsn(id, g_msn1, g_txt_Festnetz);
+addMsn(id, g_msn2, g_txt_Festnetz);
+addMsn(id, g_msn3, g_txt_Festnetz);
+addMsn(id, g_msn4, g_txt_Festnetz);
+addMsn(id, g_msn5, g_txt_Festnetz);
+addMsn(id, g_msn6, g_txt_Festnetz);
+addMsn(id, g_msn7, g_txt_Festnetz);
+addMsn(id, g_msn8, g_txt_Festnetz);
+addMsn(id, g_msn9, g_txt_Festnetz);
+addMsn(id, g_pots, g_txt_Festnetz);
+` ?> 
+// Sip Zugänge anfügen
+<? multiquery sip:settings/sip/list(ID,displayname) '
+if ("$3" != "" && isFrei("$3")) addOption(id, "$3", g_txt_Internet+g_$1);
+' ?>
+// Interne Nummern anfügen
+addMsn(id, g_int1, g_txt_Intern);
+addMsn(id, g_int2, g_txt_Intern);
+addMsn(id, g_int3, g_txt_Intern);
+addMsn(id, g_int4, g_txt_Intern);
+addMsn(id, g_int5, g_txt_Intern);
+addMsn(id, g_int6, g_txt_Intern);
+addMsn(id, g_int7, g_txt_Intern);
+addMsn(id, g_int8, g_txt_Intern);
+
+addOption(id, "Leer", "");
+jslSetSelection(id, nr=="" ? "Leer":nr);
+}
+function addMsn (id, msn, prefix) {
+if (msn != "" && isFrei(msn)) addOption(id, msn, prefix+msn);
+}
+function addOption (id, val, text) {
+var element;
+element = document.getElementById(id);
+if (element == null) return;
+//	var i=0;
+//	for (i=0; i<element.length; i++) {
+//		if (element.options[i].value == val) return; // Wert schon vorhanden
+//	}
+g_OptsCount++;
+var eintrag = new Option(text, val, false, false);
+element.options[element.length]=eintrag;
+}
+function uiIndexMSN(nr) {
+g_MsnNr = nr;
+return g_Index[Number(nr)];
+}
+</script>
--- usr/www/avme/en/html/en/fon/foneditdect.frm	2008-01-01 19:00:00.000000000 -0500
+++ usr/www/avme/en/html/en/fon/foneditdect.frm	2008-02-17 12:18:47.000000000 -0500
@@ -0,0 +1,8 @@
+<input type="hidden" name="dect:command/StartSubscription" id="uiPostStartSubscription" disabled>
+<input type="hidden" name="dect:command/Unsubscribe" id="uiPostUnsubscribe" disabled>
+<input type="hidden" name="dect:command/PageHandsets" id="uiPostPageHandsets" disabled>
+<input type="hidden" name="dect:command/DateTime" id="uiPostHandsetsTime" disabled>
+<? repeat [ 0 5 ] `
+<input type="hidden" name="dect:settings/Handset$0/Subscribed" value="<? query dect:settings/Handset$0/Subscribed ?>" id="uiPostSubscribed$0" disabled>
+<input type="hidden" name="dect:settings/Handset$0/Type" value="<? query dect:settings/Handset$0/Type ?>" id="uiPostType$0" disabled>
+` ?>
--- usr/www/avme/en/html/en/fon/foneditdect.html	2008-01-01 19:00:00.000000000 -0500
+++ usr/www/avme/en/html/en/fon/foneditdect.html	2008-02-14 14:06:37.000000000 -0500
@@ -0,0 +1,55 @@
+<? include ../html/$var:lang/fon/foneditdect.inc ?> 
+<div id="content">
+<div class="backtitel"><div class="rundrt"><div class="rundlt"><div class="ecklb"><div class="eckrb"><div class="foretitel">
+<? echo '$var:Text23' ?>
+</div></div></div></div></div></div>
+<div class="backframe"><div class="ecklm"><div class="eckrm"><div class="rundrb"><div class="rundlb"><div class="foredialog"><div class="ie_b10">
+<ul class="tabs">
+<li><a href="javascript:uiDoDect()"><? echo '$var:Text24' ?></a></li>
+<li><a href="javascript:uiDoListDect()"><? echo '$var:Text26' ?></a></li>
+<li><a href="javascript:uiDoSetupDect()"><? echo '$var:Text28' ?></a></li>
+<li class="tabs_on"><a href="javascript:uiDoCancel()"><? echo '$var:Text29' ?></a></li>
+</ul>
+<div class="backdialog"><div class="ecklm"><div class="eckrm"><div class="ecklb"><div class="eckrb"><div class="foredialog">
+<p class="mb10"><? echo $var:Text17 ?></p>
+<table id="tHandsets" class="tborder">
+<tr>
+<th class="c1"><? echo $var:Text3 ?></th>
+<th class="c2"><? echo $var:Text4 ?></th>
+<th class="c3"><? echo $var:Text20 ?></th>
+<th class="c4"></th>
+<th class="c5"></th>
+</tr>	
+<? repeat [ 0 5 ] '
+<tr id="uiViewHandset$0" style="display:none">
+<td class="c1"><option value="$0"><script type="text/javascript">document.write($0+1);</script></option></td>
+<td class="c2"><input type="checkbox" id="uiViewSubscribed$0" disabled></td>
+<td class="c3"><input type="checkbox" id="uiViewType$0" disabled></td>
+<td class="c4"></td>
+<td class="c5"><button type="button" id="uiViewDelete$0" onclick="uiDoDelete($0)" title="$var:Text31" style="width: 30px">
+<img src="../html/<? echo $var:lang ?>/images/loeschen.gif" align="bottom" width="16" height="16" hspace="4">
+</button></td>
+</tr>
+' ?>
+<tr id="uiNichts" style="display:none"><td colspan=5 class="ac"><? echo '$var:Text32' ?></td></tr>
+</table>
+<p class="mt10 ar"><input type="button" value="<? echo $var:Text21 ?>" id="uiViewNew" onclick="uiDoNew()" class="Pushbutton" style="width:180px;"></p>		
+</div></div></div></div></div></div>
+<div class="backdialog"><div class="ecklm"><div class="eckrm"><div class="ecklb"><div class="eckrb"><div class="foredialog">
+
+<p class="mb5"><? echo $var:Text35 ?></p>
+<div style="text-align: right;">
+<input type="button" value="<? echo $var:Text33 ?>" id="uiViewPage" onclick="uiDoPage()" class="Pushbutton" style="width:180px;">
+<input type="button" value="<? echo $var:Text34 ?>" id="uiViewDateTime" onclick="uiDoDateTime()" class="Pushbutton" style="width:180px;">
+<input type="button" value="<? echo $var:Text19 ?>" id="uiSendBook" onclick="uiDoSendBook()" class="Pushbutton" style="width:180px;">
+
+</div>
+</div></div></div></div></div></div>
+
+
+<div class="backdialog"><div class="ecklm"><div class="eckrm"><div class="rundrb"><div class="rundlb"><div class="forebuttons">
+<input type="button" onclick="uiDoRefresh()" value="<? echo $var:txtRefresh ?>" class=Pushbutton>
+<input type="button" onclick="uiDoHelp()" value="<? echo $var:txtHelp ?>" class="Pushbutton" id="buttonHilfe">
+</div></div></div></div></div></div>
+</div></div></div></div></div></div></div>
+</div>
--- usr/www/avme/en/html/en/fon/foneditdect.inc	2008-01-01 19:00:00.000000000 -0500
+++ usr/www/avme/en/html/en/fon/foneditdect.inc	2008-02-14 03:48:12.000000000 -0500
@@ -0,0 +1,23 @@
+<? include ../html/$var:lang/global.inc ?>
+<? setvariable var:Text3 'Number' ?>
+<? setvariable var:Text4 'Assigned' ?>
+<? setvariable var:Text7 'Change configuration' ?>
+<? setvariable var:Text16 'Assign / Unassign mobile devices' ?>
+<? setvariable var:Text17 'You can assign and unassign mobile devices to your FRITZ!Box. To assign a DECT device you need to enter the PIN (0000 - default) in you mobile device.' ?>
+<? setvariable var:Text19 'Send Phonebook' ?>
+<? setvariable var:Text20 'Swissvoice/GAP' ?>
+<? setvariable var:Text21 'Assign mobile devices' ?>
+<? setvariable var:Text22 'back' ?>
+<? setvariable var:Text23 'DECT-devices' ?>
+<? setvariable var:Text24 'Overview' ?>
+<? setvariable var:Text25 'Fixed subsriber line' ?>
+<? setvariable var:Text26 'Internal Phonenumber' ?>
+<? setvariable var:Text27 'Options' ?>
+<? setvariable var:Text28 'Assign' ?>
+<? setvariable var:Text29 'Mobile devices' ?>
+<? setvariable var:Text30 'Other devices' ?>
+<? setvariable var:Text31 'Unassign device' ?>
+<? setvariable var:Text32 'No mobile devices assigned' ?>
+<? setvariable var:Text33 'Page mobile device' ?>
+<? setvariable var:Text34 'Set date and time' ?>
+<? setvariable var:Text35 'With the Paging-Function you can locate your mobile devices. At the push of a Button you can also set Time and Date of the DECT devices.' ?>
\ No newline at end of file
--- usr/www/avme/en/html/en/fon/foneditdect.js	2008-01-01 19:00:00.000000000 -0500
+++ usr/www/avme/en/html/en/fon/foneditdect.js	2008-03-05 09:58:28.000000000 -0500
@@ -0,0 +1,89 @@
+<!--		../fon/foneditdect.js		-->
+<style type="text/css">
+<!--
+#content {width: 560px; margin: auto;}
+#tHandsets {width: 100%;}
+#tHandsets td {padding: 2px 3px;}
+#tHandsets th {padding: 2px 3px;}
+#tHandsets .c1 {text-align: center}
+#tHandsets .c2 {text-align: center}
+#tHandsets .c3 {text-align: center}
+#tHandsets .c4 {width: 0px;}
+#tHandsets .c5 {width: 30px;}
+.mt10 {margin-top: 10px;}
+.ar {text-align: right;}
+.ac {text-align: center;}
+-->
+</style>
+<script type="text/javascript">
+<? include ../html/$var:lang/js/jsl.js ?>
+var g_mldUnsubscribe	= "Do you want to unassign the mobile device ?";
+var g_New = -1;
+function uiDoOnLoad() {
+var bFound = false;
+for (var i=0; i<=8; i++) {
+if (jslGetValue("uiPostSubscribed"+i) == "1") {
+jslSetChecked("uiViewSubscribed"+i, jslGetValue("uiPostSubscribed"+i) == "1");
+jslSetChecked("uiViewType"+i, jslGetValue("uiPostType"+i) == "1");
+jslDisplay("uiViewHandset"+i, true); bFound=true;
+} else {
+jslDisable("uiViewDelete"+i)
+if (g_New == -1) g_New = i;
+}
+}
+jslDisplay("uiNichts", !bFound);
+jslSetEnabled("uiViewPage", bFound);
+if (g_New == -1) jslDisable("uiViewNew");
+jslZebrastreifen("tHandsets");
+}
+function uiDoHelp() {
+jslPopHelp("hilfe_fon_editdect");
+}
+function uiDoCancel() {
+jslGoTo("", "");
+}
+function uiDoRefresh() {
+jslGoTo("fon", "foneditdect");
+}
+function uiDoDect() {
+jslGoTo("fon", "dect0");
+}
+function uiDoListDect() {
+jslGoTo("fon", "fonlistdect");
+}
+function uiDoSetupDect() {
+jslGoTo("fon", "fonsetupdect");
+}
+function uiDoSendBook() {
+jslSetValue("uiPostPageMaster", "dect0");
+jslGoTo("fon", "buchsend");
+}
+function uiDoDelete(nr) {
+if (!confirm(g_mldUnsubscribe)) return;
+jslSetValue("uiPostUnsubscribe", nr);
+jslEnable("uiPostUnsubscribe");
+jslSubmitFormEx("fon", "", "");
+}
+function uiDoNew() {
+if (!confirm("Handteil jetzt anmelden? Pin code erforderlich!")) return;
+jslSetValue("uiPostStartSubscription", "0");
+jslEnable("uiPostStartSubscription");
+jslSubmitFormEx("fon", "", "");
+}
+function uiDoSendBook() {
+jslSetValue("uiPostPageMaster", "dect0");
+jslGoTo("fon", "buchsend");
+}
+function uiDoPage () {
+jslSetValue("uiPostPageHandsets", "0");
+jslEnable("uiPostPageHandsets");
+jslSubmitFormEx("fon", "", "");
+if (!confirm("Alle angemeldeten Handteile sollten jetzt eine Minute lang klingeln!")) return;
+}
+function uiDoDateTime () {
+jslSetValue("uiPostHandsetsTime", "0");
+jslEnable("uiPostHandsetsTime");
+jslSubmitFormEx("fon", "", "");
+if (!confirm("Uhrzeit wurde nur bei allen angemldeten Swiss-Voice comaptiblen Handteilen gesetzt!")) return;
+}
+</script>
--- usr/www/avme/en/html/en/fon/fonlist.html	2008-04-18 13:18:17.000000000 -0400
+++ usr/www/avme/en/html/en/fon/fonlist.html	2008-04-18 12:21:17.000000000 -0400
@@ -1,4 +1,4 @@
-<? include ../html/$var:lang/fon/fonlist.inc ?>
+<? include ../html/$var:lang/fon/fonlist.inc ?> 
 <div id="content">
 <div class="backtitel"><div class="rundrt"><div class="rundlt"><div class="ecklb"><div class="eckrb"><div class="foretitel">
 <? echo $var:Text1 ?>
--- usr/www/avme/en/html/en/fon/fonlistdect.frm	2008-01-01 19:00:00.000000000 -0500
+++ usr/www/avme/en/html/en/fon/fonlistdect.frm	2008-02-14 13:34:20.000000000 -0500
@@ -0,0 +1,6 @@
+<input type="hidden" name="var:IsdnNr" value="<? echo $var:IsdnNr ?>" id="uiPostIsdnNr">
+<input type="hidden" name="var:parent" value="<? echo $var:parent ?>" id="uiPostParent">
+<? repeat [ 1 8 ] `
+<input type="hidden" name="telcfg:settings/NTHotDialList/Name$0" value="<? query telcfg:settings/NTHotDialList/Name$0 ?>" id="uiPostName$0" disabled>
+<input type="hidden" name="telcfg:settings/NTHotDialList/Number$0" value="<? query telcfg:settings/NTHotDialList/Number$0 ?>" id="uiPostNummer$0" disabled>
+` ?>
--- usr/www/avme/en/html/en/fon/fonlistdect.html	2008-01-01 19:00:00.000000000 -0500
+++ usr/www/avme/en/html/en/fon/fonlistdect.html	2008-02-24 14:20:40.000000000 -0500
@@ -0,0 +1,44 @@
+<? include ../html/$var:lang/fon/fonlistdect.inc ?>
+<div id="content">
+<div class="backtitel"><div class="rundrt"><div class="rundlt"><div class="ecklb"><div class="eckrb"><div class="foretitel">
+DECT-Mobile devices
+</div></div></div></div></div></div>
+<div class="backframe"><div class="ecklm"><div class="eckrm"><div class="rundrb"><div class="rundlb"><div class="foredialog"><div class="ie_b10">
+<ul class="tabs">
+<li><a href="javascript:uiDoDect()">Overview</a></li>
+<li class="tabs_on"><a href="javascript:uiDoCancel()">Internal Number</a></li>
+<li><a href="javascript:uiDoSetupDect()">Assign</a></li>
+<li><a href="javascript:uiDoEditDect()">Mobile devices</a></li>
+</ul>
+<div class="backdialog"><div class="ecklm"><div class="eckrm"><div class="ecklb"><div class="eckrb"><div class="foredialog">
+<p class="mb10"><? echo $var:Text17 ?></p>
+<table id="tDECT" class="tborder">
+<tr>
+<th class="c1"><? echo $var:Text3 ?></th>
+<th class="c2"><? echo $var:Text4 ?></th>
+<th class="c3"><? echo $var:Text20 ?></th>
+<th class="c4"></th>
+<th class="c5"></th>
+</tr>	
+<? repeat [ 1 8 ] '
+<tr id="uiViewDECT$0" style="display:none">
+<td class="c1">5$0</td>
+<td class="c2"><? query telcfg:settings/NTHotDialList/Name$0 ?></td>
+<td class="c3"><? query telcfg:settings/NTHotDialList/Number$0 ?></td>
+<td class="c4"><button type="button" id="uiViewEdit0" onclick="uiEditDECT($0)" title="<? echo $var:txtIconBtnEdit ?>" style="width: 30px">
+<img src="../html/<? echo $var:lang ?>/images/bearbeiten.gif" align="bottom" width="16" height="16" hspace="4">
+</button></td>
+<td class="c5"><button type="button" id="uiViewDelete0" onclick="uiDeleteDECT($0)" title="<? echo $var:txtIconBtnDelete ?>" style="width: 30px">
+<img src="../html/<? echo $var:lang ?>/images/loeschen.gif" align="bottom" width="16" height="16" hspace="4">
+</button></td>
+</tr>
+' ?>
+<tr id="uiNichts" style="display:none"><td colspan=5 class="ac">There are no internal numberes assingned</td></tr>
+</table>
+<p class="mt10 ar"><input type="button" value="<? echo $var:Text21 ?>" id="uiViewNew" onclick="uiNewDECT()" class="Pushbutton" style="width:180px;"></p>
+</div></div></div></div></div></div>
+<div class="backdialog"><div class="ecklm"><div class="eckrm"><div class="rundrb"><div class="rundlb"><div class="forebuttons">
+<input type="button" onclick="uiDoHelp()" value="<? echo $var:txtHelp ?>" class="Pushbutton" id="buttonHilfe">
+</div></div></div></div></div></div>
+</div></div></div></div></div></div></div>
+</div>
--- usr/www/avme/en/html/en/fon/fonlistdect.inc	2008-01-01 19:00:00.000000000 -0500
+++ usr/www/avme/en/html/en/fon/fonlistdect.inc	2008-02-14 03:48:12.000000000 -0500
@@ -0,0 +1,19 @@
+<? include ../html/$var:lang/global.inc ?>
+<? setvariable var:Text3 'Number' ?>
+<? setvariable var:Text4 'Description' ?>
+<? setvariable var:Text7 'Change settings' ?>
+<? setvariable var:Text16 'Internal numbers for DECT devices' ?>
+<? setvariable var:Text17 'You can setup internal numbers for DECT devices. The devices can be reached via **<Nummer> from other deivces attached to your FRITZ!Box.' ?>
+<? setvariable var:Text20 'Number of the telephone' ?>
+<? setvariable var:Text31 'Delete number' ?>
+<? setvariable var:Text21 'New number' ?>
+<? setvariable var:Text22 'Back' ?>
+<? setvariable var:Text23 'DECT devices' ?
+<? setvariable var:Text24 'Overview' ?>
+<? setvariable var:Text25 'Fixed subscriber line' ?>
+<? setvariable var:Text26 'Internal numbers' ?>
+<? setvariable var:Text27 'Options' ?>
+<? setvariable var:Text28 'Assign' ?>
+<? setvariable var:Text29 'Mobile devices' ?>
+<? setvariable var:Text30 'Other devices' ?>
+
--- usr/www/avme/en/html/en/fon/fonlistdect.js	2008-01-01 19:00:00.000000000 -0500
+++ usr/www/avme/en/html/en/fon/fonlistdect.js	2008-03-05 09:59:02.000000000 -0500
@@ -0,0 +1,70 @@
+<!--		../fon/fonlistdect.js		-->
+<style type="text/css">
+<!--
+#content {width: 560px; margin: auto;}
+#tDECT {width: 100%;}
+#tDECT td {padding: 2px 3px;}
+#tDECT th {padding: 2px 3px;}
+#tDECT .c1 {text-align: center}
+#tDECT .c4, #tDECT .c5 {width: 30px;}
+.mt10 {margin-top: 10px;}
+.ar {text-align: right;}
+.ac {text-align: center;}
+-->
+</style>
+<script type="text/javascript">
+<? include ../html/$var:lang/js/jsl.js ?>
+var g_mldDelete	= "Do you want to delete this number ?";
+var g_New = 0;
+function uiDoOnLoad() {
+var bFound = false;
+for (var i=1; i<=8; i++) {
+if (jslGetValue("uiPostNummer"+i) != "") {
+jslDisplay("uiViewDECT"+i, true); bFound=true;
+} else {
+if (g_New == 0) g_New = i;
+}
+}
+jslDisplay("uiNichts", !bFound);
+if (g_New == 0) jslDisable("uiViewNew");
+jslZebrastreifen("tDECT");
+
+}
+function uiDoHelp() {
+jslPopHelp("hilfe_fon_listdect");
+}
+function uiDoCancel() {
+jslGoTo("", "");
+}
+function uiDoRefresh() {
+jslGoTo("fon", "fonelistdect");
+}
+function uiDoDect() {
+jslGoTo("fon", "dect0");
+}
+function uiDoEditDect() {
+jslGoTo("fon", "foneditdect");
+}
+function uiDoSetupDect() {
+jslGoTo("fon", "fonsetupdect");
+}
+
+function uiEditDECT(nr) {
+jslCopyValue("uiPostParent", "uiPostPageName");
+jslSetValue("uiPostIsdnNr", nr);
+jslSubmitFormEx("fon", "fon1isdn", jslGetValue("uiPostPageName"));
+}
+function uiDeleteDECT(nr) {
+if (!confirm(g_mldDelete)) return;
+jslSetValue("uiPostName"+nr, "");
+jslSetValue("uiPostNummer"+nr, "");
+jslEnable("uiPostName"+nr);
+jslEnable("uiPostNummer"+nr);
+jslSubmitFormEx("fon", "", "");
+}
+function uiNewDECT() {
+jslCopyValue("uiPostParent", "uiPostPageName");
+jslSetValue("uiPostIsdnNr", g_New);
+jslSubmitFormEx("fon", "fon1isdn", jslGetValue("uiPostPageName"));
+}
+</script>
--- usr/www/avme/en/html/en/fon/fonsetupdect.frm	2008-01-01 19:00:00.000000000 -0500
+++ usr/www/avme/en/html/en/fon/fonsetupdect.frm	2008-02-14 13:34:20.000000000 -0500
@@ -0,0 +1,9 @@
+<input type="hidden" name="var:MsnNr" value="<? echo $var:MsnNr ?>" id="uiPostMsnNr">
+<input type="hidden" name="dect:settings/RingOnAllMSNs" value="<? query dect:settings/RingOnAllMSNs ?>" id="uiPostRingOnAllMSNs" disabled>
+<? repeat [ 0 4 ] `
+<input type="hidden" name="dect:settings/MSN$0/MSN" value="<? query dect:settings/MSN$0/MSN ?>" id="uiPostMSN$0" disabled>
+<input type="hidden" name="dect:settings/MSN$0/CLIR" value="<? query dect:settings/MSN$0/CLIR ?>" id="uiPostCLIR$0" disabled>
+<input type="hidden" name="dect:settings/MSN$0/CallWaiting" value="<? query dect:settings/MSN$0/CallWaiting ?>" id="uiPostCallWaiting$0" disabled>
+<input type="hidden" name="dect:settings/MSN$0/BusyOnBusy" value="<? query dect:settings/MSN$0/BusyOnBusy ?>" id="uiPostBusyOnBusy$0" disabled>
+<input type="hidden" name="dect:settings/MSN$0/Handsets" value="<? query dect:settings/MSN$0/Handsets ?>" id="uiPostHandsets$0" disabled>
+` ?>
--- usr/www/avme/en/html/en/fon/fonsetupdect.html	2008-01-01 19:00:00.000000000 -0500
+++ usr/www/avme/en/html/en/fon/fonsetupdect.html	2008-02-14 13:34:20.000000000 -0500
@@ -0,0 +1,54 @@
+<? include ../html/$var:lang/fon/fonsetupdect.inc ?> 
+<div id="content">
+<div class="backtitel"><div class="rundrt"><div class="rundlt"><div class="ecklb"><div class="eckrb"><div class="foretitel">
+<? echo '$var:Text23' ?>
+</div></div></div></div></div></div>
+<div class="backframe"><div class="ecklm"><div class="eckrm"><div class="rundrb"><div class="rundlb"><div class="foredialog"><div class="ie_b10">
+<ul class="tabs">
+<li><a href="javascript:uiDoDect()"><? echo '$var:Text24' ?></a></li>
+<li><a href="javascript:uiDoListDect()"><? echo '$var:Text26' ?></a></li>
+<li class="tabs_on"><a href="javascript:uiDoCancel()"><? echo '$var:Text28' ?></a></li>
+<li><a href="javascript:uiDoEditDect()"><? echo '$var:Text29' ?></a></li>
+</ul>
+<div class="backdialog"><div class="ecklm"><div class="eckrm"><div class="ecklb"><div class="eckrb"><div class="foredialog">
+<p class="mb10"><? echo $var:Text1 ?></p>
+<p class="mb10"><input type="checkbox" id="uiViewRingOnAllMSNs" onclick="uiRingOnAllMSNs()">&nbsp;<label for="uiViewRingOnAllMSNs"><? echo $var:Text2 ?></label></p>
+</div></div></div></div></div></div>
+<div class="backdialog"><div class="ecklm"><div class="eckrm"><div class="ecklb"><div class="eckrb"><div class="foredialog">
+<p class="mb10"><? echo $var:Text17 ?></p>
+<table id="tMSN" class="tborder">
+<tr>
+<th class="c1"><? echo $var:Text3 ?></th>
+<th class="c2"><? echo $var:Text4 ?></th>
+<th class="c3"><? echo $var:Text5 ?></th>
+<th class="c4"><? echo $var:Text6 ?></th>
+<th class="c5"><? echo $var:Text7 ?></th>
+<th class="c6"><? echo $var:Text8 ?></th>
+<th class="c7"></th>
+<th class="c8"></th>
+</tr>	
+<? repeat [ 0 4 ] '
+<tr id="uiViewMSN$0" style="display:none">
+<td class="c1"><option value="$0"><script type="text/javascript">document.write(uiIndexMSN("$0"));</script></option></td>
+<td class="c2"><? query dect:settings/MSN$0/MSN ?></td>
+<td class="c3"><input type="checkbox" id="uiViewCLIR$0" disabled></td>
+<td class="c4"><input type="checkbox" id="uiViewCallWaiting$0" disabled></td>
+<td class="c5"><input type="checkbox" id="uiViewBusyOnBusy$0" disabled></td>
+<td class="c6"><script type="text/javascript">document.write(uiListHandsets("<? query dect:settings/MSN$0/Handsets ?>"));</script></td>
+<td class="c7"><button type="button" id="uiViewEdit$0" onclick="uiEditMSN($0)" title="$var:Text32" style="width: 30px">
+<img src="../html/<? echo $var:lang ?>/images/bearbeiten.gif" align="bottom" width="16" height="16" hspace="4">
+</button></td>
+<td class="c8"><button type="button" id="uiViewDelete$0" onclick="uiDeleteMSN($0)" title="$var:Text31" style="width: 30px">
+<img src="../html/<? echo $var:lang ?>/images/loeschen.gif" align="bottom" width="16" height="16" hspace="4">
+</button></td>
+</tr>
+' ?>
+<tr id="uiNichts" style="display:none"><td colspan=5 class="ac">Es sind keine MSNs eingerichtet</td></tr>
+</table>
+<p class="mt10 ar"><input type="button" value="<? echo $var:Text21 ?>" id="uiViewNew" onclick="uiNewMSN()" class="Pushbutton" style="width:180px;"></p>
+</div></div></div></div></div></div>
+<div class="backdialog"><div class="ecklm"><div class="eckrm"><div class="rundrb"><div class="rundlb"><div class="forebuttons">
+<input type="button" onclick="uiDoHelp()" value="<? echo $var:txtHelp ?>" class="Pushbutton" id="buttonHilfe">
+</div></div></div></div></div></div>
+</div></div></div></div></div></div></div>
+</div>
--- usr/www/avme/en/html/en/fon/fonsetupdect.inc	2008-01-01 19:00:00.000000000 -0500
+++ usr/www/avme/en/html/en/fon/fonsetupdect.inc	2008-02-14 03:48:13.000000000 -0500
@@ -0,0 +1,26 @@
+<? include ../html/$var:lang/global.inc ?>
+<? setvariable var:Text1 'Mobile devices can be setup to recognize all incoming calls. If setting this option all assigned numbers will be lost.' ?>
+<? setvariable var:Text2 'React on all numbers' ?>
+<? setvariable var:Text3 'MSN' ?>
+<? setvariable var:Text4 'Phonenumber' ?>
+<? setvariable var:Text5 'Anonymous' ?>
+<? setvariable var:Text6 'Knock' ?>
+<? setvariable var:Text7 'MSN busy' ?>
+<? setvariable var:Text8 'Mobile devices' ?>
+<? setvariable var:Text17 'Assign up to 5 numbers to the mobile devices, calls can be taken and calls can be placed on.' ?>
+<? setvariable var:Text21 'New Assignment' ?>
+<? setvariable var:Text23 'DECT devices' ?>
+<? setvariable var:Text24 'Overview' ?>
+<? setvariable var:Text25 'Fixed subscriber line' ?>
+<? setvariable var:Text26 'Internal numbers' ?>
+<? setvariable var:Text27 'Options' ?>
+<? setvariable var:Text28 'Assign' ?>
+<? setvariable var:Text29 'Mobile devices' ?>
+<? setvariable var:Text30 'Other devices' ?>
+<? setvariable var:Text32 'Change Assignment' ?>
+<? setvariable var:Text31 'Delete Assignment' ?>
+<? setvariable var:Text50 'A' ?>
+<? setvariable var:Text51 'B' ?>
+<? setvariable var:Text52 'C' ?>
+<? setvariable var:Text53 'D' ?>
+<? setvariable var:Text54 'E' ?>
\ No newline at end of file
--- usr/www/avme/en/html/en/fon/fonsetupdect.js	2008-01-01 19:00:00.000000000 -0500
+++ usr/www/avme/en/html/en/fon/fonsetupdect.js	2008-03-05 09:59:37.000000000 -0500
@@ -0,0 +1,125 @@
+<!--		../fon/fonsetupdect.js		-->
+<style type="text/css">
+<!--
+#content {width: 560px; margin: auto;}
+#tMSN {width: 100%;}
+#tMSN td {padding: 2px 3px;}
+#tMSN th {padding: 2px 3px;}
+#tMSN .c1 {text-align: center}
+#tMSN .c2 {text-align: left}
+#tMSN .c3 {text-align: center}
+#tMSN .c4 {text-align: center}
+#tMSN .c5 {text-align: center}
+#tMSN .c6 {text-align: left}
+#tMSN .c7, #tMSN .c8 {width: 30px;}
+.mt10 {margin-top: 10px;}
+.ar {text-align: right;}
+.ac {text-align: center;}
+-->
+</style>
+<script type="text/javascript">
+<? include ../html/$var:lang/js/jsl.js ?>
+var g_mldDelete	= "Delete assignment ?";
+var g_mldAllOn	= "Delete all assignments ?";
+var g_New = -1;
+var g_Sequence = new Array("1 ", "2 ", "3 ", "4 ", "5 ", "6 ");
+var g_Index = new Array("A", "B", "C", "D", "E");
+var g_Handsets;
+function uiDoOnLoad() {
+g_Handsets = 0;
+
+<? multiquery dect:settings/Handset/list(Subscribed) `
+<? if eq $2 1 `
+g_Handsets |= (1<<$0);
+` ?>
+` ?>
+
+jslSetChecked("uiViewRingOnAllMSNs", jslGetValue("uiPostRingOnAllMSNs") == "1");
+jslSetEnabled("uiViewNew", jslGetValue("uiPostRingOnAllMSNs") != "1");
+
+var bFound = false;
+for (var i=0; i<5; i++) {
+if (jslGetValue("uiPostMSN"+i) != "") {
+jslSetChecked("uiViewCLIR"+i, jslGetValue("uiPostCLIR"+i) == "1");
+jslSetChecked("uiViewCallWaiting"+i, jslGetValue("uiPostCallWaiting"+i) == "1");
+jslSetChecked("uiViewBusyOnBusy"+i, jslGetValue("uiPostBusyOnBusy"+i) == "1");
+jslDisplay("uiViewMSN"+i, true); bFound=true;
+} else {
+jslDisable("uiViewEdit"+i)
+jslDisable("uiViewDelete"+i)
+if (g_New == -1) g_New = i;
+}
+}
+jslDisplay("uiNichts", !bFound);
+if (g_New == -1) jslDisable("uiViewNew");
+jslZebrastreifen("tMSN");
+}
+function uiDoHelp() {
+jslPopHelp("hilfe_fon_setupdect");
+}
+function uiDoCancel() {
+jslGoTo("", "");
+}
+function uiDoRefresh() {
+jslGoTo("fon", "fonsetupdect");
+}
+function uiDoDect() {
+jslGoTo("fon", "dect0");
+}
+function uiDoListDect() {
+jslGoTo("fon", "fonlistdect");
+}
+function uiDoEditDect() {
+jslGoTo("fon", "foneditdect");
+}
+function uiEditMSN(nr) {
+jslCopyValue("uiPostPageMaster", "uiPostPageName");
+jslSetValue("uiPostMsnNr", nr);
+jslSubmitFormEx("fon", "fon1dect", jslGetValue("uiPostPageName"));
+}
+function uiDeleteMSN(nr) {
+if (!confirm(g_mldDelete)) return;
+jslSetValue("uiPostMSN"+nr, "");
+jslEnable("uiPostMSN"+nr);
+jslSubmitFormEx("fon", "", "");
+}
+function uiNewMSN() {
+jslCopyValue("uiPostPageMaster", "uiPostPageName");
+jslSetValue("uiPostMsnNr", g_New);
+jslSubmitFormEx("fon", "fon1dect", jslGetValue("uiPostPageName"));
+}
+function uiRingOnAllMSNs() {
+var bFound = false;
+if (jslGetChecked("uiViewRingOnAllMSNs")) {
+for (var j=0; j<5; j++) {
+if (jslGetValue("uiPostMSN"+j) != "") bFound=true;
+}
+if (bFound) {
+if (!confirm(g_mldAllOn)) {
+jslSetChecked("uiViewRingOnAllMSNs", false);
+return;
+} 
+for (var i=0; i<5; i++) {
+jslSetValue("uiPostMSN"+i, "");
+jslEnable("uiPostMSN"+i);
+}
+}
+}
+jslSetValue("uiPostRingOnAllMSNs", jslGetChecked("uiViewRingOnAllMSNs") ? "1":"0");
+jslEnable("uiPostRingOnAllMSNs");
+jslSubmitFormEx("fon", "", "");
+}
+function uiIndexMSN(nr) {
+return g_Index[Number(nr)];
+}
+function uiListHandsets(bitset) {
+var bitset = (bitset.charCodeAt(0) - 48);
+var bits = Number(bitset);
+var list = "";
+for (var j=0; j<6; j++) {
+//		if (bits & g_Handsets & (1<<j) != 0)	list += g_Sequence[j];
+if ((bits & (1<<j)) != 0)	list += g_Sequence[j];
+}
+return list;
+}
+</script>
