Index: speed-to-fritz/sp-to-fritz.sh
===================================================================
--- speed-to-fritz/sp-to-fritz.sh	(revision 1116)
+++ speed-to-fritz/sp-to-fritz.sh	(working copy)
@@ -15,7 +15,7 @@
 # Date of current version:
 # TODO: LC_ALL= LANG= LC_TIME= svn info . | awk '/^Last Changed Date: / {print $4}'
 #dont chang this line formwat is used in ./start to get script version into Firmware.conf
-Tag="13"; Monat="10"; Jahr="10"
+Tag="14"; Monat="10"; Jahr="10"
 export SKRIPT_DATE="$Tag.$Monat.$Jahr"
 export SKRIPT_DATE_ISO="$Jahr.$Monat.$Tag"
 export SKRIPT_REVISION="$Jahr$Monat$Tag"
Index: speed-to-fritz/Config.in
===================================================================
--- speed-to-fritz/Config.in	(revision 1116)
+++ speed-to-fritz/Config.in	(working copy)
@@ -3955,15 +3955,26 @@
 	default n
 	help
 	    Box MOD ID should be left as it is, but if needed you may specify your own!.
- 	config  EXPORT_FBMOD
-	    depends on SET_FBMOD
-	    string "Type in FritzBox Mod ID, see help for more info."
-	    default ""
-	    help
-		This is the ID name used instead of the ID used in AVM firmware, 
-		use "7270, 7170, ...  but can be any valid ID used in download addresses.
-		This ID is used to construct the URLs for "Programme" and for plugin download. 
 
+ config  EXPORT_FBMOD
+	depends on SET_FBMOD
+	string "Type in FritzBox Mod ID, see help for more info."
+	default ""
+	help
+	    This is the ID name used instead of the ID used in AVM firmware, 
+	    use "7270, 7170, ...  but can be any valid ID used in download addresses.
+	    This ID is used to construct the URLs for "Programme" and for plugin download. 
+
+ config SET_NAME
+	bool "Own product name (setup)"
+	depends on EXPERT_MOD
+	default n
+	help
+            Product name should be left as it is, but if needed you may specify your own.
+            Some tools like Fritz!FAX may be not usable if the box product name has been changed!
+            This is not the Host name of your box! This is a string transparent on the net
+            to other programs and on the settings page of the Speedport.
+
  config  EXPORT_NEWNAME
 	    depends on SET_NAME
 	    string "product name"
