Index: co/henrynestler/start.sh
===================================================================
--- co/henrynestler/start.sh	(revision 112)
+++ co/henrynestler/start.sh	(working copy)
@@ -4,6 +4,19 @@
 # revision 11
 #http://docs.blackfin.uclinux.org/doku.php?id=colinux:building-installer&s[]=colinux
 # Remove left over Subversion directories
+echo "-------------------------------------------------------------------------------------------------------------"
+echo
+if ! [ `id -u` -eq 0 ]; then
+ clear
+  echo
+  echo "This script needs to be executed with 'su' privileges."
+  echo 
+  echo "Use ./s.sh instead!"
+  echo "!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!"
+  sleep 10
+  exit 0
+fi
+echo "-------------------------------------------------------------------------------------------------------------"
 echo "====================================================================================================="
 echo "====================================================================================================="
 echo "Checking for left over Subversion directories"
Index: co/henrynestler/s.sh
===================================================================
--- co/henrynestler/s.sh	(revision 0)
+++ co/henrynestler/s.sh	(revision 0)
@@ -0,0 +1,14 @@
+#!/bin/bash
+echo "-------------------------------------------------------------------------------------------------------------"
+echo
+if [ `id -u` -eq 0 ]; then
+ clear
+  echo
+  echo "This script needs to be executed without 'su' privileges."
+  echo "!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!"
+  sleep 10
+  exit 0
+fi
+echo "-------------------------------------------------------------------------------------------------------------"
+sudo sh -c './start.sh'
+sleep 20
\ No newline at end of file

Property changes on: co/henrynestler/s.sh
___________________________________________________________________
Name: svn:executable
   + *

