Index: speed-to-fritz/Config.in
===================================================================
--- speed-to-fritz/Config.in	(revision 1036)
+++ speed-to-fritz/Config.in	(working copy)
@@ -220,9 +220,9 @@
     string
     default "@AVM/fritz.box/fritzbox.fon_wlan_7240/firmware/deutsch/FRITZ.Box_Fon_WLAN_7240.73.04.80.image"
 
-config AVM_PRE_7240_18158
+config AVM_PRE_7240_18202
     string
-    default "http://www.avm.de/de/Service/Service-Portale/Labor/7240_vorschau_release_candidate/labor_start_release_candidate.php HW144:fritzbox-labor-7240-18158.zip ./Firmware.orig/Labor_FRITZ.Box_Fon_WLAN_7240.73.04.85-18158.image"
+    default "http://www.avm.de/de/Service/Service-Portale/Labor/7240_vorschau_release_candidate/labor_start_release_candidate.php HW144:fritzbox-labor-7240-18202.zip ./Firmware.orig/Labor_FRITZ.Box_Fon_WLAN_7240.73.04.85-18202.image"
 ###------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 config AVM_7272_54_04_58
     string
@@ -288,14 +288,14 @@
 #    string
 #    default "ftp://update.avm.de/labor/ipv6/FRITZ.Box_Fon_WLAN_7270_16_LabIPV6.54.04.80-17603.image mirror select"
 
-config AVM_PRE_7271_18158
+config AVM_PRE_7271_18202
     string
-    default "http://www.avm.de/de/Service/Service-Portale/Labor/7270_vorschau_release_candidate/labor_start_vorschau_release_candidate.php HW122:fritzbox-labor-7270v1-18158.ZIP ./Firmware.orig/Labor_FRITZ.Box_Fon_WLAN_7270_v1.54.04.85-18158.image"
+    default "http://www.avm.de/de/Service/Service-Portale/Labor/7270_vorschau_release_candidate/labor_start_vorschau_release_candidate.php HW122:fritzbox-labor-7270v1-18202.zip ./Firmware.orig/Labor_FRITZ.Box_Fon_WLAN_7270_v1.54.04.85-18202.image"
 
 # 7272 is used instead of 7270v2
-config AVM_PRE_7272_18158
+config AVM_PRE_7272_18202
     string
-    default "http://www.avm.de/de/Service/Service-Portale/Labor/7270_vorschau_release_candidate/labor_start_vorschau_release_candidate.php HW139:fritzbox-labor-7270v2-18158.ZIP ./Firmware.orig/Labor_FRITZ.Box_Fon_WLAN_7270_v2.54.04.85-18158.image"
+    default "http://www.avm.de/de/Service/Service-Portale/Labor/7270_vorschau_release_candidate/labor_start_vorschau_release_candidate.php HW139:fritzbox-labor-7270v2-18202.zip ./Firmware.orig/Labor_FRITZ.Box_Fon_WLAN_7270_v2.54.04.85-18202.image"
 # Muster eines Links zur php Seite, HW ID muss zusaetzlich angegeben werden damit der richtige Link vom Skript auf der Seite gefunden werden kann. Die Zeile wird automatisch beim Skriptdurchlauf vervollstaendigt.   
 #    default "http://www.avm.de/de/Service/Service-Portale/Labor/7270_vorschau_release_candidate/labor_start_vorschau_release_candidate.php HWxxx:"
 
@@ -307,9 +307,9 @@
     string
     default "ftp://update.avm.de/labor/RC/Labor_FRITZ.Box_Fon_WLAN_7270_v2.54.04.85-18158.image mirror select"
 
-config AVM_PRE_7273_18158
+config AVM_PRE_7273_18202
    string
-    default "http://www.avm.de/de/Service/Service-Portale/Labor/7270_vorschau_release_candidate/labor_start_vorschau_release_candidate.php HW145:fritzbox-labor-7270v3-18158.zip ./Firmware.orig/Labor_FRITZ.Box_Fon_WLAN_7270_v3.74.04.85-18158.image"
+    default "http://www.avm.de/de/Service/Service-Portale/Labor/7270_vorschau_release_candidate/labor_start_vorschau_release_candidate.php HW145:fritzbox-labor-7270v3-18202.zip ./Firmware.orig/Labor_FRITZ.Box_Fon_WLAN_7270_v3.74.04.85-18202.image"
 config AVM_DSL_7271_16750
     string
     default "http://www.avm.de/de/Service/Service-Portale/Labor/7270_dsl/labor_start_dsl.php HW122:"
@@ -408,9 +408,9 @@
    string
      default "@AVM/fritz.box/fritzbox.fon_wlan_7390/firmware/deutsch/fritz.box_fon_wlan_7390_84.04.84.image"
 
-config AVM_PRE_7390_18158
+config AVM_PRE_7390_18202
    string
-     default "http://www.avm.de/de/Service/Service-Portale/Labor/7390_vorschau_release_candidate/labor_start_neues_aus_der_entwicklung.php HW156:fritzbox-labor-7390-18158.zip ./Firmware.orig/FRITZ.Box_Fon_WLAN_7390.Labor.84.04.85-18158.image"
+     default "http://www.avm.de/de/Service/Service-Portale/Labor/7390_vorschau_release_candidate/labor_start_neues_aus_der_entwicklung.php HW156:fritzbox-labor-7390-18202.zip ./Firmware.orig/FRITZ.Box_Fon_WLAN_7390.Labor.84.04.85-18202.image"
 
 ### -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 
@@ -1079,10 +1079,10 @@
 		default TYPE_7272_54_04_59 if TYPE_SPEEDPORT_W503V
 		default TYPE_7170_58_04_80 if ( ! TYPE_SPECIALW721_TCOM && TYPE_SPEEDPORT_W721V )
 		default TYPE_AIO_7272_14543 if ! EXPORT_TYPE_LOCAL_MODEL
-#		default TYPE_PRE_7272_18158 if TYPE_OR_7570_920 && ! FORCE_VDSL
-#		default TYPE_PRE_7272_18158 if TYPE_OR_7570_920 && FORCE_VDSL
+#		default TYPE_PRE_7272_18202 if TYPE_OR_7570_920 && ! FORCE_VDSL
+#		default TYPE_PRE_7272_18202 if TYPE_OR_7570_920 && FORCE_VDSL
 		default TYPE_PRE_7272_17828 if TYPE_OR_7570_920
-		default TYPE_PRE_7273_18158 if TYPE_7240
+		default TYPE_PRE_7273_18202 if TYPE_7240
 		#->(only for testing)
 		default TYPE_7390_PRE_18101 if TYPE_7390
 #		default TYPE_7390_84_04_84 if TYPE_7390
@@ -1232,8 +1232,8 @@
 		help
 			FRITZ! release: 80
 
-	config TYPE_PRE_7240_18158
-		bool "AVM 7240 18158 (PREVIEW)"
+	config TYPE_PRE_7240_18202
+		bool "AVM 7240 18202 (PREVIEW)"
 		depends on ( TYPE_OR_7240_7570_503_722_920_AVMX_LOCALTCOM ) && ( TYPE_OR_7570XX_DE )
 		help
 			FRITZ!Lab (PREVIEW):
@@ -1257,14 +1257,14 @@
 		help
 			FRITZ!Lab (PREVIEW):
 
-	config TYPE_PRE_7271_18158
-		bool "AVM 7270v1 18158 (PREVIEW)"
+	config TYPE_PRE_7271_18202
+		bool "AVM 7270v1 18202 (PREVIEW)"
 		depends on ( TYPE_OR_7240_7570_503_722_920_AVMX_LOCALTCOM ) && ( TYPE_OR_7570XX_DE )
 		help
 			FRITZ!Lab (PREVIEW):
 			
-	config TYPE_PRE_7272_18158
-		bool "AVM 7270v2 18158 (PREVIEW)"
+	config TYPE_PRE_7272_18202
+		bool "AVM 7270v2 18202 (PREVIEW)"
 		depends on ( TYPE_OR_7240_7570_503_722_920_AVMX_LOCALTCOM ) && ( TYPE_OR_7570XX_DE )
 		help
 			FRITZ!Lab (PREVIEW):
@@ -1290,8 +1290,8 @@
 #		help
 #			FRITZ! release: 76
 
-	config TYPE_PRE_7273_18158
-		bool "AVM 7270v3 18158 (PREVIEW)"
+	config TYPE_PRE_7273_18202
+		bool "AVM 7270v3 18202 (PREVIEW)"
 		depends on ( TYPE_OR_7240_7273_7390_7570_503_722_920_AVMX_LOCALTCOM ) && ( TYPE_OR_7570XX_DE )
 		help 
 			FRITZ!Lab (PREVIEW):
@@ -1688,8 +1688,8 @@
 	config TYPE_SRC3_AVM_7240_73_04_80
 		bool "AVM 7240 73.04.80"
 
-	config TYPE_SRC3_AVM_PRE_7240_18158
-		bool "AVM 7240 18158 (PREVIEW)"
+	config TYPE_SRC3_AVM_PRE_7240_18202
+		bool "AVM 7240 18202 (PREVIEW)"
 
 ###-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 ### 7270v2 Firmware
@@ -1781,17 +1781,17 @@
 	config TYPE_SRC3_AVM_PRE_7272_17828
 		bool "AVM 7270v2 17828 (PREVIEW)"
 
-	config TYPE_SRC3_AVM_PRE_7271_18158
-		bool "AVM 7270v1 18158 (PREVIEW)"
+	config TYPE_SRC3_AVM_PRE_7271_18202
+		bool "AVM 7270v1 18202 (PREVIEW)"
 
-	config TYPE_SRC3_AVM_PRE_7272_18158
-		bool "AVM 7270v2 18158 (PREVIEW)"
+	config TYPE_SRC3_AVM_PRE_7272_18202
+		bool "AVM 7270v2 18202 (PREVIEW)"
 
 	config TYPE_SRC3_AVM_PRE_72XX
 		bool "AVM 72XX (PREVIEW), select one"
 
-	config TYPE_SRC3_AVM_PRE_7273_18158
-		bool "AVM 7270v3 18158 (PREVIEW)"
+	config TYPE_SRC3_AVM_PRE_7273_18202
+		bool "AVM 7270v3 18202 (PREVIEW)"
 
 ###-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 ### 7390 Firmware
@@ -1805,8 +1805,8 @@
 	config TYPE_SRC3_AVM_PRE_7390_17370
 		bool "AVM 7390 17370"
 
-	config TYPE_SRC3_AVM_PRE_7390_18158
-		bool "AVM 7390 18158"
+	config TYPE_SRC3_AVM_PRE_7390_18202
+		bool "AVM 7390 18202"
 
 ###-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 ### 7570 Firmware
@@ -1947,7 +1947,7 @@
 	default "$AVM_7273_74_04_80"		if TYPE_7273
 #	default "$AVM_7390_84_04_82"		if TYPE_7390
 #	default "$AVM_7390_84_04_84"		if TYPE_7390
-	default "$AVM_PRE_7390_18158"		if TYPE_7390
+	default "$AVM_PRE_7390_18202"		if TYPE_7390
 	default "$TCOM_LOCAL"	 			if TYPE_LOCAL_TCOM
 	default "$TCOM_7570_LOCAL"	 		if TYPE_LOCAL_TCOM_7570
 
@@ -2022,11 +2022,11 @@
 	default "$AVM_NAS_7273_16561"				if ( TYPE_NAS_7273_16561 ) && ( TYPE_OR_7570_503_722_920 || EXPORT_TYPE_LOCAL_MODEL )
 	default "$AVM_PHONE_72XX"				if ( TYPE_PHONE_72XX) && ( TYPE_OR_7570_503_722_920 || EXPORT_TYPE_LOCAL_MODEL )
 	default "$AVM_PRE_72XX"					if ( TYPE_PRE_72XX) && ( TYPE_OR_7570_503_722_920 || EXPORT_TYPE_LOCAL_MODEL )
-	default "$AVM_PRE_7271_18158"				if ( TYPE_PRE_7271_18158 ) && ( TYPE_OR_7570_503_722_920 || EXPORT_TYPE_LOCAL_MODEL )
-	default "$AVM_PRE_7273_18158"				if ( TYPE_PRE_7273_18158 ) && ( TYPE_OR_7570_503_722_920 || EXPORT_TYPE_LOCAL_MODEL )
-	default "$AVM_PRE_7272_18158"				if ( TYPE_PRE_7272_18158 ) && ( TYPE_OR_7570_503_722_920 || EXPORT_TYPE_LOCAL_MODEL )
+	default "$AVM_PRE_7271_18202"				if ( TYPE_PRE_7271_18202 ) && ( TYPE_OR_7570_503_722_920 || EXPORT_TYPE_LOCAL_MODEL )
+	default "$AVM_PRE_7273_18202"				if ( TYPE_PRE_7273_18202 ) && ( TYPE_OR_7570_503_722_920 || EXPORT_TYPE_LOCAL_MODEL )
+	default "$AVM_PRE_7272_18202"				if ( TYPE_PRE_7272_18202 ) && ( TYPE_OR_7570_503_722_920 || EXPORT_TYPE_LOCAL_MODEL )
 	default "$AVM_PRE_7272_17828"				if ( TYPE_PRE_7272_17828 ) && ( TYPE_OR_7570_503_722_920 || EXPORT_TYPE_LOCAL_MODEL )
-	default "$AVM_PRE_7240_18158"				if ( TYPE_PRE_7240_18158 ) && ( TYPE_OR_7570_503_722_920 || EXPORT_TYPE_LOCAL_MODEL )
+	default "$AVM_PRE_7240_18202"				if ( TYPE_PRE_7240_18202 ) && ( TYPE_OR_7570_503_722_920 || EXPORT_TYPE_LOCAL_MODEL )
 	default "$AVM_7570_75_04_78"				if ( TYPE_AVM_7570_75_04_78 ) && ( TYPE_OR_7570_503_722_920 )
 	default "$AVM_7570_75_04_82"				if ( TYPE_AVM_7570_75_04_82 ) && ( TYPE_OR_7570_503_722_920 )
 	default "$AVM_7570_15245"					if ( TYPE_AVM_7570_15245 ) && ( TYPE_OR_7570_503_722_920 )
@@ -2039,7 +2039,7 @@
 	default "$AVM_7390_84_04_81"					if TYPE_7390_84_04_81
 	default "$AVM_7390_84_04_82"					if TYPE_7390_84_04_82
 	default "$AVM_7390_84_04_84"					if TYPE_7390_84_04_84
-	default "$AVM_PRE_7390_18158"					if TYPE_7390_PRE_18101
+	default "$AVM_PRE_7390_18202"					if TYPE_7390_PRE_18101
 	default "$TCOM_W722_80_04_76" 					if TYPE_W722_80_04_76
 #<--
 ###-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
@@ -2101,12 +2101,12 @@
 	default "$AVM_DSL_7271_16750"			if TYPE_SRC3_AVM_DSL_7271_16750
 	default "$AVM_DSL_7272_16750"			if TYPE_SRC3_AVM_DSL_7272_16750
 	default "$AVM_DSL_7273_16750"			if TYPE_SRC3_AVM_DSL_7273_16750
-	default "$AVM_PRE_7240_18158"			if TYPE_SRC3_AVM_PRE_7240_18158
+	default "$AVM_PRE_7240_18202"			if TYPE_SRC3_AVM_PRE_7240_18202
 	default "$AVM_PRE_7272_17828"			if TYPE_SRC3_AVM_PRE_7272_17828
-	default "$AVM_PRE_7271_18158"			if TYPE_SRC3_AVM_PRE_7271_18158
-	default "$AVM_PRE_7272_18158"			if TYPE_SRC3_AVM_PRE_7272_18158
+	default "$AVM_PRE_7271_18202"			if TYPE_SRC3_AVM_PRE_7271_18202
+	default "$AVM_PRE_7272_18202"			if TYPE_SRC3_AVM_PRE_7272_18202
 	default "$AVM_PRE_72XX"				if TYPE_SRC3_AVM_PRE_72XX
-	default "$AVM_PRE_7273_18158"			if TYPE_SRC3_AVM_PRE_7273_18158
+	default "$AVM_PRE_7273_18202"			if TYPE_SRC3_AVM_PRE_7273_18202
 	default "$AVM_PHONE_7240_16376"			if TYPE_SRC3_AVM_PHONE_7240_16376
 	default "$AVM_PHONE_7272_16999"			if TYPE_SRC3_AVM_PHONE_7272_16999
 	default "$AVM_PHONE_7273_16999"			if TYPE_SRC3_AVM_PHONE_7273_16999
@@ -2115,7 +2115,7 @@
 	default "$AVM_7390_84_04_81"			if TYPE_SRC3_AVM_PRE_7390_15936
 	default "$AVM_7390_84_04_82"			if TYPE_SRC3_AVM_PRE_7390_16165
 	default "$AVM_7390_84_04_84"			if TYPE_SRC3_AVM_PRE_7390_17370
-	default "$AVM_PRE_7390_18158"			if TYPE_SRC3_AVM_PRE_7390_18158
+	default "$AVM_PRE_7390_18202"			if TYPE_SRC3_AVM_PRE_7390_18202
 	default "$AVM_PHONE_72XX"				if TYPE_SRC3_AVM_PHONE_72XX
 	default "$AVM_7570_75_04_78" 			if TYPE_SRC3_AVM_7570_75_04_78
 	default "$AVM_7570_75_04_82" 			if TYPE_SRC3_AVM_7570_75_04_82
@@ -2150,21 +2150,21 @@
 ###-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 config TYPE_7271_TYPE
 	bool
-	default "y"	 	if  TYPE_CORE_7271_16540 || TYPE_DSL_7271_16750 || TYPE_PRE_7271_18158 
+	default "y"	 	if  TYPE_CORE_7271_16540 || TYPE_DSL_7271_16750 || TYPE_PRE_7271_18202 
 
 config TYPE_7272_TYPE
 	bool
-	default "y"	 	if  TYPE_CORE_7272_16540 || TYPE_DSL_7272_16750 || TYPE_PRE_7272_18158 ||\
+	default "y"	 	if  TYPE_CORE_7272_16540 || TYPE_DSL_7272_16750 || TYPE_PRE_7272_18202 ||\
 				TYPE_7272_73_04_76 || TYPE_7272_73_04_80
 
 config TYPE_7273_TYPE
 	bool
-	default "y"	 	if  TYPE_CORE_7273_15220 || TYPE_CORE_7273_16540 || TYPE_PHONE_7273_16999 || TYPE_IPV6_7273_17603 || TYPE_PRE_7273_18158 ||\
+	default "y"	 	if  TYPE_CORE_7273_15220 || TYPE_CORE_7273_16540 || TYPE_PHONE_7273_16999 || TYPE_IPV6_7273_17603 || TYPE_PRE_7273_18202 ||\
 				TYPE_PRE_72XX || TYPE_PHONE_72XX || TYPE_IPV6_72XX || TYPE_7273_73_04_76 || TYPE_7273_73_04_80
 
 config TYPE_V3_OR_7240
 	bool
-	default "y" 		if  TYPE_7273_TYPE || TYPE_7240_73_04_80 || TYPE_CORE_7240_16540 || TYPE_CORE_7240_16540 || TYPE_PHONE_7240_16376 || TYPE_PRE_7240_18158
+	default "y" 		if  TYPE_7273_TYPE || TYPE_7240_73_04_80 || TYPE_CORE_7240_16540 || TYPE_CORE_7240_16540 || TYPE_PHONE_7240_16376 || TYPE_PRE_7240_18202
 
 config TYPE_LABOR_TYPE
 	string
@@ -2177,7 +2177,7 @@
 	default "70" 		if TYPE_7170_70 || TYPE_7272_54_04_70
 	default "76m" 		if TYPE_7272_m54_04_76
 	default "CORE"	 	if TYPE_CORE_7273_15220 || TYPE_CORE_7272_15220 || TYPE_CORE_7271_16540 || TYPE_CORE_7272_16540 || TYPE_CORE_7273_16540 || TYPE_CORE_7240_16540 || TYPE_7240_CORE_16540
-	default "PREVIEW"	if TYPE_PRE_7240_18158 || TYPE_PRE_7272_17828 || TYPE_PRE_7271_18158 || TYPE_PRE_7272_18158 || TYPE_PRE_7273_18158 || TYPE_7170_PRE_16352 || TYPE_7390_PRE_18101 || TYPE_PRE_72XX
+	default "PREVIEW"	if TYPE_PRE_7240_18202 || TYPE_PRE_7272_17828 || TYPE_PRE_7271_18202 || TYPE_PRE_7272_18202 || TYPE_PRE_7273_18202 || TYPE_7170_PRE_16352 || TYPE_7390_PRE_18101 || TYPE_PRE_72XX
 #	default "13948" 	if TYPE_7170_13948
 #	default "13486" 	if TYPE_7272_13486
 	default "NAS"		if TYPE_NAS_7272_16561 || TYPE_NAS_7273_16561
