Index: fwmod
===================================================================
--- fwmod	(working copy)
+++ fwmod	(working copy)
@@ -167,7 +161,12 @@
 UNSQUASHFS_TOOL="unsquashfs3-lzma"
 EXTERNAL_TOOL="external"
 
-MKSQUASHFS_OPTIONS="-le -noappend -all-root -info"
+MKSQUASHFS_OPTIONS="-noappend -all-root -info"
+if [ "$FREETZ_TARGET_ARCH" = "mips" ]; then 
+    MKSQUASHFS_OPTIONS+=" -be"
+        else 
+    MKSQUASHFS_OPTIONS+=" -le"
+        fi --- fwmod	(revision 4819)
 UNSQUASHFS_OPTIONS="-no-progress"
 
 TAR_TOOL="tar"
