Index: Config.in
===================================================================
--- Config.in	(revision 4819)
+++ Config.in	(working copy)
@@ -2844,14 +2844,59 @@
 		bool "Build toolchain (requires 4GB diskspace)"
 endchoice
 
+config FREETZ_TARGET_ARCH_LE
+	bool
+	default y
+	depends on \
+		FREETZ_TYPE_2170 || \
+		FREETZ_TYPE_300IP_AS_FON || \
+		FREETZ_TYPE_FON || \
+		FREETZ_TYPE_FON_5010 || \
+		FREETZ_TYPE_FON_5050 || \
+		FREETZ_TYPE_FON_5140 || \
+		FREETZ_TYPE_FON_5224 || \
+		FREETZ_TYPE_FON_WLAN || \
+		FREETZ_TYPE_FON_WLAN_7050 || \
+		FREETZ_TYPE_FON_WLAN_7112 || \
+		FREETZ_TYPE_FON_WLAN_7113 || \
+		FREETZ_TYPE_FON_WLAN_7141 || \
+		FREETZ_TYPE_FON_WLAN_7140 || \
+		FREETZ_TYPE_FON_7150 || \
+		FREETZ_TYPE_FON_WLAN_7170 || \
+		FREETZ_TYPE_FON_WLAN_7240 || \
+		FREETZ_TYPE_FON_WLAN_7270 || \
+		FREETZ_TYPE_FON_WLAN_7270_V3 || \
+		FREETZ_TYPE_FON_WLAN_7570 || \
+		FREETZ_TYPE_WLAN_3020 || \
+		FREETZ_TYPE_WLAN_3030 || \
+		FREETZ_TYPE_WLAN_3130 || \
+		FREETZ_TYPE_WLAN_3131 || \
+		FREETZ_TYPE_WLAN_3170 || \
+		FREETZ_TYPE_WLAN_3270 || \
+		FREETZ_TYPE_WLAN_3270_V3 || \
+		FREETZ_TYPE_SPEEDPORT_W501V
+
+#config FREETZ_TARGET_ARCH_BE
+#	bool
+#	default y
+#	depends on \
+#		FREETZ_TYPE_FON_WLAN_7390
+
 config FREETZ_TARGET_ARCH
 	string
-	default "mipsel"
+	default "mips" if FREETZ_TARGET_ARCH_BE
+	default "mipsel" if FREETZ_TARGET_ARCH_LE
 
 config FREETZ_TARGET_CROSS
 	string
-	default "mipsel-linux-uclibc-"
+	default "mips-linux-uclibc-" if FREETZ_TARGET_ARCH_BE
+	default "mipsel-linux-uclibc-" if FREETZ_TARGET_ARCH_LE
 
+config FREETZ_MKSQUASHFS_TARGET_ARCH_OPTION
+	string
+	default "-be" if FREETZ_TARGET_ARCH_BE
+	default "-le" if FREETZ_TARGET_ARCH_LE
+
 config FREETZ_TARGET_MAKE_PATH
 	string
 	default "toolchain/target/bin"
@@ -2868,7 +2913,8 @@
 
 config FREETZ_KERNEL_CROSS
 	string
-	default "mipsel-unknown-linux-gnu-"
+	default "mips-unknown-linux-gnu-" if FREETZ_TARGET_ARCH_BE
+	default "mipsel-unknown-linux-gnu-" if FREETZ_TARGET_ARCH_LE
 
 config FREETZ_KERNEL_MAKE_PATH
 	string
Index: fwmod
===================================================================
--- fwmod	(revision 4819)
+++ fwmod	(working copy)
@@ -167,7 +167,7 @@
 UNSQUASHFS_TOOL="unsquashfs3-lzma"
 EXTERNAL_TOOL="external"
 
-MKSQUASHFS_OPTIONS="-le -noappend -all-root -info"
+MKSQUASHFS_OPTIONS="$FREETZ_MKSQUASHFS_TARGET_ARCH_OPTION -noappend -all-root -info"
 UNSQUASHFS_OPTIONS="-no-progress"
 
 TAR_TOOL="tar"
