diff -Nur make/linux/kernel.mk 16MB-freetz-make/linux/kernel.mk
--- make/linux/kernel.mk	2010-04-26 18:43:57.782155000 +0200
+++ make/linux/kernel.mk	2010-04-27 20:49:38.000000000 +0200
@@ -176,7 +176,7 @@
 		. | tar -xf - -C $(KERNEL_MODULES_DIR)
 	touch $@
 
-kernel-precompiled: pkg-echo-start $(KERNEL_TARGET_DIR)/$(KERNEL_TARGET_BINARY) $(KERNEL_MODULES_DIR)/.modules-$(KERNEL_LAYOUT) pkg-echo-done
+kernel-precompiled: pkg-echo-start pkg-echo-done
 
 kernel-configured: $(KERNEL_DIR)/.depend_done
