--- usr/www/all/html/de/home/foncalls.js.orig	2009-09-15 15:26:24.000000000 +0200
+++ usr/www/all/html/de/home/foncalls.js	2009-09-16 19:22:16.000000000 +0200
@@ -56,7 +56,7 @@
 var g_sym1 = "<img src=\"../html/de/images/Callin.gif\" title=\"{?de.home.foncalls.js:40?}\">";
 var g_sym2 = "<img src=\"../html/de/images/Callinfailed.gif\" title=\"{?de.home.foncalls.js:45?}\">";
 var g_sym3 = "<img src=\"../html/de/images/Callout.gif\" title=\"{?de.home.foncalls.js:50?}\">";
-var g_sym0 = "<img src=\"../html/de/images/Callno.gif\">";
+var g_sym0 = "<img src=\"../html/de/images/Callno.gif\" style=\"vertical-align:text-top\">";
 var g_Fon1 = "{?de.home.foncalls.js:55?}";
 var g_Fon2 = "{?de.home.foncalls.js:60?}";
 var g_Fon3 = "{?de.home.foncalls.js:65?}";
@@ -351,6 +351,9 @@
 return buchentryinfo;
 }
 }
+function uiRufnummerInfo (nr) {
+    return "<a href=\"http://www.dasoertliche.de/Controller?form_name=search_inv&ph=$(urlencode "+nr+")\" target=\"_blank\" title=\"Rückwärtssuche bei dasoertliche.de nach "+nr+"\"><img src=\"../html/<? echo $var:lang ?>/images/bearbeiten.gif\" style=\"vertical-align:text-top\"></a>";
+}
 function uiNummerDisplay (nr, name,telbooknumbertyp) {
 var buchname = GetBuchName(name);
 var buchnrtype = GetBuchNumberType(telbooknumbertyp);
@@ -358,16 +361,16 @@
 var tooltip=nr;
 if (nr!="")
 tooltip=buchentryinfo+" = "+nr;
-if (nr=="" && buchname=="") return g_txtUnbekannt;
+if (nr=="" && buchname=="") return g_sym0 + g_txtUnbekannt;
 if ("<? query telcfg:settings/UseClickToDial ?>" == "1") {
-if (nr == "") return span(buchentryinfo);
+if (nr == "") return g_sym0 + span(buchentryinfo);
 if (buchname == "") {
 return "<nobr>" +
-"<a href=\"javascript:Dial('"+nr+"')\" title=\""+nr+"\">"+nr+"</a></nobr>";
+uiRufnummerInfo(nr) + "<a href=\"javascript:Dial('"+nr+"')\" title=\""+nr+"\">"+nr+"</a></nobr>";
 }
-return "<nobr><a href=\"javascript:Dial('"+nr+"')\" title=\""+tooltip+"\">"+buchentryinfo+"</a></nobr>";
+return "<nobr>" +  g_sym0 + "<a href=\"javascript:Dial('"+nr+"')\" title=\""+tooltip+"\">"+buchentryinfo+"</a></nobr>";
 } else {
-return spanEx(tooltip,buchname=="" ? nr:buchentryinfo);
+return (buchname=="" ? uiRufnummerInfo(nr):g_sym0) + spanEx(tooltip,buchname=="" ? nr:buchentryinfo);
 }
 }
 function uiRouteDisplay (n, t) {
