--- etc/hotplug/run_mount	2010-01-29 06:31:40.000000000 -0500
+++ etc/hotplug/run_mount	2010-03-14 15:20:49.000000000 -0400
@@ -6,6 +6,9 @@
 FTPGID=0
 MSTART=/sbin/start_mediasrv
 WEBDAVCONTROL=/etc/webdav_control
+ # Do we have a Ftp server (ftpd) or was ist stripped from the firmware?
+_ftpd=$(basename $(which ftpd) 2> /dev/null)
+ ## Wait for removal of filesystem semaphores (mask: $1, maximal wait period: $2)
 TR069START=/bin/tr069starter
 SYSFS=/var/sysfs
 ## Generate a frontend'able 'nice name' to be used as mount point:
@@ -107,9 +110,11 @@
 FAIL_EVENT=0
 umask $OLD_UMASK
 eventadd 140 $MNTNAME
+if [ "$_ftpd" ] ; then
 if test -x /etc/ftpd_control ; then
 /etc/ftpd_control reconfig
 fi
+fi
 test -x $TR069START && $TR069START $MNTNAME
 if test -x /etc/samba_control ; then
 /etc/samba_control reconfig
--- etc/hotplug/storage	2010-01-29 06:31:43.000000000 -0500
+++ etc/hotplug/storage	2010-03-14 15:24:38.000000000 -0400
@@ -3,6 +3,8 @@
 DEVMAP=/var/media/devmap
 FTPDIR=/var/media/ftp
 SYSFS=/var/sysfs
+# Do we have a Ftp server (ftpd) or was ist stripped from the firmware?
+_ftpd=$(basename $(which ftpd) 2> /dev/null)
 TIMEOUT=30
 FTPUID=0
 FTPGID=0
@@ -74,9 +76,11 @@
 if test -x /etc/samba_control; then
 /etc/samba_control reconfig "$MPOINT"
 fi
+if [ "$_ftpd" ] ; then
 if test -x /etc/ftpd_control; then
 /etc/ftpd_control reconfig
 fi
+fi
 return $ERROR
 }
 do_umount () {
