Index: toolchain/Config.in
===================================================================
--- toolchain/Config.in	(revision 4818)
+++ toolchain/Config.in	(working copy)
@@ -6,7 +6,8 @@
 			FREETZ_TYPE_FON_WLAN_7240 || \
 			FREETZ_TYPE_FON_WLAN_7270 || \
 			FREETZ_TYPE_FON_WLAN_7270_V3 || \
-			FREETZ_TYPE_FON_WLAN_7570 \
+			FREETZ_TYPE_FON_WLAN_7570  || \
+			FREETZ_TYPE_FON_WLAN_7390 \
 			)
 	default n
 
@@ -39,9 +40,9 @@
 		FREETZ_TYPE_FON_WLAN_7240 || \
 		FREETZ_TYPE_FON_WLAN_7270 || \
 		FREETZ_TYPE_FON_WLAN_7270_V3 || \
-		FREETZ_TYPE_FON_WLAN_7570
+		FREETZ_TYPE_FON_WLAN_7570 || \
+		FREETZ_TYPE_FON_WLAN_7390
 	default n
-
 config FREETZ_TARGET_UCLIBC_VERSION_0_9_30
 	bool
 	default n
diff -Naur Config.in Config.in
--- Config.in	2010-04-30 10:06:07.000000000 -0400
+++ Config.in	2010-04-30 10:01:49.000000000 -0400
@@ -28,6 +28,8 @@
 	bool
 config FREETZ_AVM_VERSION_7270_04_80
 	bool
+config FREETZ_AVM_VERSION_7390_04_83
+	bool
 config FREETZ_AVM_VERSION_r7203
 	bool
 
@@ -119,6 +121,12 @@
 		select FREETZ_AVM_VERSION_7270_04_80
 		bool "Fon WLAN 7570 VDSL"
 
+	config FREETZ_TYPE_FON_WLAN_7390
+		select FREETZ_AVM_VERSION_7390_04_83
+		#TODO: download toolchain
+		select FREETZ_BUILD_TOOLCHAIN
+		bool "Fon WLAN 7390"
+
 	config FREETZ_TYPE_WLAN_3020
 		select FREETZ_AVM_VERSION_04_33
 		bool "WLAN 3020"
@@ -324,6 +332,7 @@
 	default "04.80"			if FREETZ_AVM_VERSION_04_80
 	default "3270_04.80"		if FREETZ_AVM_VERSION_3270_04_80
 	default "7270_04.80"		if FREETZ_AVM_VERSION_7270_04_80
+	default "7390_04.83"		if FREETZ_AVM_VERSION_7390_04_83
 	default "r7203"			if FREETZ_AVM_VERSION_r7203
 
 choice
@@ -427,6 +436,7 @@
 	default "7270_v3_labor_nas"				if FREETZ_TYPE_FON_WLAN_7270_V3 && FREETZ_TYPE_LABOR_NAS
 	default "7270_v3_labor_phone"				if FREETZ_TYPE_FON_WLAN_7270_V3 && FREETZ_TYPE_LABOR_PHONE
 	default "7570"						if FREETZ_TYPE_FON_WLAN_7570
+	default "7390"						if FREETZ_TYPE_FON_WLAN_7390
 	default "W501V"						if FREETZ_TYPE_SPEEDPORT_W501V
 	default "custom"					if FREETZ_TYPE_CUSTOM
 
@@ -488,7 +498,7 @@
 	depends on \
 		FREETZ_SHOW_ADVANCED && \
 		FREETZ_REPLACE_KERNEL_AVAILABLE
-	select FREETZ_MODULE_fuse if (FREETZ_AVM_VERSION_3270_04_80 || FREETZ_AVM_VERSION_7270_04_80)
+	select FREETZ_MODULE_fuse if (FREETZ_AVM_VERSION_3270_04_80 || FREETZ_AVM_VERSION_7270_04_80 || FREETZ_AVM_VERSION_7390_04_83 )
 	default n
 	help
 		Replace AVM kernel with self-built kernel.
@@ -520,6 +530,8 @@
 
 config FREETZ_TARGET_REF
 	string "Target ref"		if FREETZ_TYPE_CUSTOM
+	default "16mb_26"		if \
+		FREETZ_TYPE_FON_WLAN_7390
 	default "8mb_26"		if \
 		FREETZ_TYPE_FON_5124 || \
 		FREETZ_TYPE_FON_WLAN_7112 || \
@@ -552,6 +564,8 @@
 
 config FREETZ_KERNEL_REF
 	string "Kernel ref"		if FREETZ_TYPE_CUSTOM
+	default "16mb_26"		if \
+		FREETZ_TYPE_FON_WLAN_7390
 	default "8mb_26"		if \
 		FREETZ_TYPE_FON_5124 || \
 		FREETZ_TYPE_FON_WLAN_7112 || \
@@ -614,6 +628,8 @@
 		FREETZ_TYPE_FON_WLAN_7570 || \
 		FREETZ_TYPE_WLAN_3270 || \
 		FREETZ_TYPE_WLAN_3270_V3
+	default "iks"		if \
+		FREETZ_TYPE_FON_WLAN_7390
 
 config FREETZ_KERNEL_MTD_SIZE
 	int "Kernel (64K blocks)"	if FREETZ_TYPE_CUSTOM
@@ -649,7 +665,8 @@
 		FREETZ_TYPE_WLAN_3270_V3 || \
 		FREETZ_TYPE_FON_WLAN_7270_16MB || \
 		FREETZ_TYPE_FON_WLAN_7270_V3 || \
-		FREETZ_TYPE_FON_WLAN_7570
+		FREETZ_TYPE_FON_WLAN_7570 || \
+		FREETZ_TYPE_FON_WLAN_7390
 	help
 		Number of 64K blocks in the kernel mtd device.
 
@@ -1910,6 +1927,7 @@
 	default "@AVM/fritz.box/fritzbox.fon_wlan_7170/x_misc/opensrc"			if	FREETZ_AVM_VERSION_04_80
 	default "@AVM/fritz.box/fritzbox.wlan_3270/x_misc/opensrc"			if	FREETZ_AVM_VERSION_3270_04_80
 	default "@AVM/fritz.box/fritzbox.fon_wlan_7270/x_misc/opensrc"			if	FREETZ_AVM_VERSION_7270_04_80
+	default "@AVM/fritz.box/fritzbox.fon_wlan_7390/x_misc/opensrc"			if	FREETZ_AVM_VERSION_7390_04_83
 	default "http://www.t-home.de/dlp/eki/downloads/Speedport/Speedport_W501V"	if	FREETZ_AVM_VERSION_r7203
 
 config FREETZ_DL_KERNEL_SOURCE
@@ -1925,6 +1943,7 @@
 	default "fritzbox7170-source-files-04.80.tar.gz"	if	FREETZ_AVM_VERSION_04_80
 	default "fritz_box_wlan_3270_source _files .04.80.tar.gz" if	FREETZ_AVM_VERSION_3270_04_80
 	default "fritzbox7270-source-files-04.80.tar.gz"	if	FREETZ_AVM_VERSION_7270_04_80
+	default "fritz_box_fon_wlan_7390_source_files.04.83.tar.gz" if	FREETZ_AVM_VERSION_7390_04_83
 	default "GPL-r7203-4mb_26-tar.bz2"			if	FREETZ_AVM_VERSION_r7203
 
 config FREETZ_DL_KERNEL_SOURCE_MD5
@@ -1940,6 +1959,7 @@
 	default "6bf92b81b48a3a05efd3aae6c05fe3e2"		if	FREETZ_AVM_VERSION_04_80
 	default "a7e605a9f27bf514b232ec9976b9ff98"		if	FREETZ_AVM_VERSION_3270_04_80
 	default "bcd49b28a54293e1481b8170b07245e0"		if	FREETZ_AVM_VERSION_7270_04_80
+	default "c714d1707954bef7a15172fd47468306"		if	FREETZ_AVM_VERSION_7390_04_83
 	default "582c74f0959a687c41c1bcfa599ace9c"		if	FREETZ_AVM_VERSION_r7203
 
 config FREETZ_DL_SITE
@@ -2009,6 +2029,8 @@
 	default "@AVM/fritz.box/fritzbox.fon_wlan_7270_v3/firmware/deutsch"	if	FREETZ_TYPE_FON_WLAN_7270_V3 && \
 											FREETZ_TYPE_LANG_DE
 	default "@AVM/fritz.box/fritzbox.fon_wlan_7570/firmware/english"	if	FREETZ_TYPE_FON_WLAN_7570
+	default "@AVM/fritz.box/fritzbox.fon_wlan_7390/firmware/deutsch"	if	FREETZ_TYPE_FON_WLAN_7390 && \
+											FREETZ_TYPE_LANG_DE
 	default "@AVM/fritz.box/fritzbox.sl_wlan/firmware"			if	FREETZ_TYPE_WLAN_3020
 	default "@AVM/fritz.box/fritzbox.wlan_3030/firmware"			if	FREETZ_TYPE_WLAN_3030
 	default "@AVM/fritz.box/fritzbox.wlan_3130/firmware"			if	FREETZ_TYPE_WLAN_3130
@@ -2108,6 +2130,7 @@
 	default "FRITZ.Box_Fon_WLAN_7270_v3_Labor_74.04.80-16624.image"		if	FREETZ_TYPE_FON_WLAN_7270_V3 && \
 											FREETZ_TYPE_LABOR_PHONE
 	default "FRITZ.Box_Fon_WLAN_7570_vDSL.en-de-fr.75.04.81.image"		if	FREETZ_TYPE_FON_WLAN_7570
+	default "fritz.box_fon_wlan_7390_84.04.83.image"			if	FREETZ_TYPE_FON_WLAN_7390
 	default "fritz.box_sl_wlan.09.04.34.image"				if	FREETZ_TYPE_WLAN_3020
 	default "fritz.box_wlan_3030.21.04.34.image"				if	FREETZ_TYPE_WLAN_3030
 	default "fritz.box_wlan_3130.44.04.34.image"				if	FREETZ_TYPE_WLAN_3130
@@ -2164,6 +2187,7 @@
 	default "12fb134d01f6bd5e99cdab2104184380"	if	FREETZ_TYPE_FON_WLAN_7270_V3 && ! FREETZ_TYPE_LABOR && \
 								FREETZ_TYPE_LANG_DE
 	default "6cfef46ebae2c6d173e266d8f75b72f7"	if	FREETZ_TYPE_FON_WLAN_7570
+	default "c714d1707954bef7a15172fd47468306"	if	FREETZ_TYPE_FON_WLAN_7390
 	default "f483cfab8da97207248f5e0703993a72"	if	FREETZ_TYPE_WLAN_3020
 	default "31653fb1ba044375ab389474951c1cf4"	if	FREETZ_TYPE_WLAN_3030
 	default "1c8dcee29f84c96648aa4b96271c9a16"	if	FREETZ_TYPE_WLAN_3130
@@ -2876,11 +2900,11 @@
 		FREETZ_TYPE_WLAN_3270_V3 || \
 		FREETZ_TYPE_SPEEDPORT_W501V
 
-#config FREETZ_TARGET_ARCH_BE
-#	bool
-#	default y
-#	depends on \
-#		FREETZ_TYPE_FON_WLAN_7390
+config FREETZ_TARGET_ARCH_BE
+	bool
+	default y
+	depends on \
+		FREETZ_TYPE_FON_WLAN_7390
 
 config FREETZ_TARGET_ARCH
 	string
