--- /dslsnrset.js	1969-12-31 19:00:00.000000000 -0500
+++ /dslsnrset.js	2009-04-02 05:48:21.000000000 -0400
@@ -0,0 +1,194 @@
+﻿<!-- ../internet/dslsnrset.js -->
+<STYLE type = "text/css">
+<!--
+.SlideDiv{position:relative;top:0px;left:0px;width: 680px; height: 50px;}
+.SNR_td_tadio{
+width:20px;
+}
+.SNR_td_tadio_text{
+width:120px;
+}
+.SNRRadio{
+height:12px;
+}
+.SNRSetTable{
+}
+.RadioConnectStrich{
+height:1px;
+width:70px;
+color:black;
+background-color:black;
+margin:0px;
+padding:0px;
+font-size:1px;
+}
+.RadioConnectStrichTd{
+height:1px;
+}
+.OSNR_td_tadio{
+width:80px;
+}
+.OSNRRadio{
+height:12px;
+}
+.OSNRSetTable{
+margin-top:0px;
+}
+-->
+</STYLE>
+<script type="text/javascript">
+<? include ../html/de/js/jsl.js ?>
+<? include ../html/de/internet/awatch.js ?>
+var g_mode=0;
+var g_oem = "<? query env:status/OEM ?>";
+<? if eq $var:isMultiAnnex 1 `
+var g_mldChangeAnnex = "{?de.internet.dslsnrset.js:8335?}";
+var g_mldChangeOver = "{?de.internet.dslsnrset.js:5311?}";
+g_mldChangeAnnexQuestion=g_mldChangeAnnex;
+` ?>
+<? if eq $var:isMultiAnnex 1 `
+var g_mldChangeDSLMode= "{?de.internet.dslsnrset.js:1460?}";
+` ?>
+function uiDoOnLoad() {
+document.title ="{?de.internet.dslsnrset.js:5SNRPage?}";
+InitMode();
+jslSetChecked("uiViewDefaultSNR"+g_mode, true);
+<? if eq $var:isMultiAnnex 1 `
+InitAnnex();
+` ?>
+<? if eq $var:isMultiAnnex 1 `
+InitDSLMode();
+` ?>
+}
+<? if eq $var:isMultiAnnex 1 `
+function InitAnnex()
+{
+annex = jslGetValue("uiPostAnnex");
+if (annex=="A")
+uiSetAnnex(0);
+else if (annex=="B")
+uiSetAnnex(1);
+}
+function uiSetAnnex(n)
+{
+jslSetChecked("uiViewAnnexA", n==0);
+jslSetChecked("uiViewAnnexB", n==1);
+}
+` ?>
+<? if eq $var:isMultiAnnex 1 `
+function InitDSLMode()
+{
+dslmode = jslGetValue("uiPostDSLMode");
+if (dslmode=="0")
+uiSetDSLMode(0);
+else if (dslmode=="1")
+uiSetDSLMode(1);
+}
+function uiSetDSLMode(n)
+{
+jslSetChecked("uiViewDSLModeAuto", n==0);
+jslSetChecked("uiViewDSLModeVDSLOnly", n==1);
+<? if eq $var:isMultiAnnex 1 `
+if(n==0){
+g_mldChangeAnnexQuestion=g_mldChangeAnnex;
+jslSetSpanText('uiViewAnnexALabel','{?de.internet.dslsnrset.js:4120?}');
+jslSetSpanText('uiViewAnnexBLabel','{?de.internet.dslsnrset.js:7359?}');
+jslSetSpanText('uiAnnexHead','{?de.internet.dslsnrset.js:2588?}');
+}else if(n==1){
+g_mldChangeAnnexQuestion=g_mldChangeOver;
+jslSetSpanText('uiViewAnnexALabel','{?de.internet.dslsnrset.js:7197?}');
+jslSetSpanText('uiViewAnnexBLabel','{?de.internet.dslsnrset.js:8710?}');
+jslSetSpanText('uiAnnexHead','{?de.internet.dslsnrset.js:8759?}');
+}else{
+}
+` ?>
+}
+` ?>
+function InitMode(){
+var ctlbits = jslGetValue("uiPostControlBitfield");
+if(ctlbits & 4){
+snrmargin=(jslGetValue("iPostDownstreamMarginOffset")/10).toString(10);
+switch(snrmargin){
+case "0":g_mode=0;break;
+case "1":g_mode=1;break;
+case "2":g_mode=2;break;
+case "3":g_mode=3;break;
+case "4":g_mode=4;break;
+default: g_mode=0;break;
+}
+}else{
+g_mode=0;
+}
+}
+function OnChangeMode(mode){
+g_mode=mode;
+}
+function uiDoRefresh() {
+jslGoTo("internet","dslsnrset");
+}
+function uiDoHelp() {
+jslPopHelp("hilfe_internet_dslsnrset");
+}
+function uiDoSave() {
+<? if eq $var:isMultiAnnex 1 `
+bAnnexConfirm=false;
+annex = jslGetValue("uiPostAnnex");
+if (jslGetChecked("uiViewAnnexA") && annex!="A") {
+if (confirm(g_mldChangeAnnexQuestion)) {
+jslSetValue("uiPostAnnex", "A");
+jslEnable("uiPostAnnex");
+bAnnexConfirm=true;
+}
+}
+if (jslGetChecked("uiViewAnnexB") && annex!="B") {
+if (confirm(g_mldChangeAnnexQuestion)) {
+jslSetValue("uiPostAnnex", "B");
+jslEnable("uiPostAnnex");
+bAnnexConfirm=true;
+}
+}
+` ?>
+bDSLModeConfirm=false;
+dlsmode = jslGetValue("uiPostDSLMode");
+if (jslGetChecked("uiViewDSLModeAuto") && dlsmode!="0") {
+if (confirm(g_mldChangeDSLMode)) {
+jslSetValue("uiPostDSLMode", "0");
+jslEnable("uiPostDSLMode");
+bDSLModeConfirm=true;
+}
+}
+if (jslGetChecked("uiViewDSLModeVDSLOnly") && dlsmode!="1") {
+if (confirm(g_mldChangeDSLMode)) {
+jslSetValue("uiPostDSLMode", "1");
+jslEnable("uiPostDSLMode");
+bDSLModeConfirm=true;
+}
+}
+jslSetValue("iPostDownstreamMarginOffset", g_mode * 10);
+var ctlbits = jslGetValue("uiPostControlBitfield");
+ctlbits &= ~6;
+if(g_mode>0)
+{
+ctlbits |= 4;
+}
+jslSetValue("uiPostControlBitfield", ctlbits);
+if (
+<? if eq $var:isMultiAnnex 1 `
+bAnnexConfirm
+|| bDSLModeConfirm
+` `
+0
+|| bDSLModeConfirm
+` ?>
+){
+jslSetValue("uiPostGetPage", "../html/reboot.html");
+document.getElementById("uiPostForm").submit();
+}
+else {
+jslFormSubmitEx("internet", "dslsnrset");
+}
+}
+function uiDoCancel() {
+jslGoTo("internet", "dslsnrset");
+}
+</script>
